// @schedule-x/vue/dist/schedule-x-vue.mjs
import { h as Q, defineComponent as ft, isReactive as Jt, Teleport as Zt, Fragment as qt, watch as Xt, onMounted as Qt } from "vue";
const en = (e, t) => (n, i) => {
  const r = {
    Component: Q(t, i),
    wrapperElement: n
  };
  e(r);
};
class tn extends Error {
  constructor(t) {
    super(t), this.name = "[Schedule-X reactivity error]";
  }
}
const la = ft({
  name: "ScheduleXCalendar",
  props: {
    calendarApp: {
      type: Object,
      required: !0
    },
    customComponents: {
      type: Object,
      default: () => ({})
    }
  },
  data() {
    return {
      elId: "sx" + Math.random().toString(36).substr(2, 9),
      customComponentsMeta: []
    };
  },
  mounted() {
    if (Jt(this.calendarApp))
      throw new tn("calendarApp cannot be saved in a ref. Since this causes deep reactivity, it destroys the calendars internal reactivity. Save in a normal const or shallowRef");
    const e = {
      ...this.customComponents,
      ...this.$slots
    };
    for (const [t, n] of Object.entries(e))
      n && this.calendarApp._setCustomComponentFn(t, en(this.setCustomComponentMeta, n));
    this.calendarApp.render(document.getElementById(this.elId));
  },
  unmounted() {
    this.calendarApp.destroy();
  },
  methods: {
    setCustomComponentMeta(e) {
      if (!(e.wrapperElement instanceof HTMLElement))
        return;
      const n = ({ wrapperElement: a }) => a instanceof HTMLElement, i = [
        ...this.customComponentsMeta.filter(n)
      ], r = e.wrapperElement.dataset.ccid, o = i.find(({ wrapperElement: a }) => a.dataset.ccid === r);
      o && i.splice(i.indexOf(o), 1), this.customComponentsMeta = [...i, e];
    }
  },
  render() {
    const e = this.customComponentsMeta.map(({ Component: t, wrapperElement: n }) => Q(Zt, { to: n }, t));
    return Q("div", {
      id: this.elId,
      class: "sx-vue-calendar-wrapper"
    }, Q(qt, {}, e));
  }
});
var ve, h, ht, pt, V, Ke, vt, be, mt, We, we, Ye, kt, te = {}, Dt = [], nn = /acit|ex(?:s|g|n|p|$)|rph|grid|ows|mnc|ntw|ine[ch]|zoo|^ord|itera/i, me = Array.isArray;
function $(e, t) {
  for (var n in t) e[n] = t[n];
  return e;
}
function $e(e) {
  e && e.parentNode && e.parentNode.removeChild(e);
}
function H(e, t, n) {
  var i, r, o, a = {};
  for (o in t) o == "key" ? i = t[o] : o == "ref" ? r = t[o] : a[o] = t[o];
  if (arguments.length > 2 && (a.children = arguments.length > 3 ? ve.call(arguments, 2) : n), typeof e == "function" && e.defaultProps != null) for (o in e.defaultProps) a[o] === void 0 && (a[o] = e.defaultProps[o]);
  return ue(e, a, i, r, null);
}
function ue(e, t, n, i, r) {
  var o = { type: e, props: t, key: n, ref: i, __k: null, __: null, __b: 0, __e: null, __c: null, constructor: void 0, __v: r ?? ++ht, __i: -1, __u: 0 };
  return r == null && h.vnode != null && h.vnode(o), o;
}
function Y(e) {
  return e.children;
}
function A(e, t) {
  this.props = e, this.context = t;
}
function z(e, t) {
  if (t == null) return e.__ ? z(e.__, e.__i + 1) : null;
  for (var n; t < e.__k.length; t++) if ((n = e.__k[t]) != null && n.__e != null) return n.__e;
  return typeof e.type == "function" ? z(e) : null;
}
function yt(e) {
  var t, n;
  if ((e = e.__) != null && e.__c != null) {
    for (e.__e = e.__c.base = null, t = 0; t < e.__k.length; t++) if ((n = e.__k[t]) != null && n.__e != null) {
      e.__e = e.__c.base = n.__e;
      break;
    }
    return yt(e);
  }
}
function Ne(e) {
  (!e.__d && (e.__d = !0) && V.push(e) && !_e.__r++ || Ke !== h.debounceRendering) && ((Ke = h.debounceRendering) || vt)(_e);
}
function _e() {
  var e, t, n, i, r, o, a, s;
  for (V.sort(be); e = V.shift(); ) e.__d && (t = V.length, i = void 0, o = (r = (n = e).__v).__e, a = [], s = [], n.__P && ((i = $({}, r)).__v = r.__v + 1, h.vnode && h.vnode(i), Fe(n.__P, i, r, n.__n, n.__P.namespaceURI, 32 & r.__u ? [o] : null, a, o ?? z(r), !!(32 & r.__u), s), i.__v = r.__v, i.__.__k[i.__i] = i, gt(a, i, s), i.__e != o && yt(i)), V.length > t && V.sort(be));
  _e.__r = 0;
}
function Mt(e, t, n, i, r, o, a, s, u, d, _) {
  var l, c, f, M, m, S, D = i && i.__k || Dt, y = t.length;
  for (u = rn(n, t, D, u, y), l = 0; l < y; l++) (f = n.__k[l]) != null && (c = f.__i === -1 ? te : D[f.__i] || te, f.__i = l, S = Fe(e, f, c, r, o, a, s, u, d, _), M = f.__e, f.ref && c.ref != f.ref && (c.ref && Re(c.ref, null, f), _.push(f.ref, f.__c || M, f)), m == null && M != null && (m = M), 4 & f.__u || c.__k === f.__k ? u = Pt(f, u, e) : typeof f.type == "function" && S !== void 0 ? u = S : M && (u = M.nextSibling), f.__u &= -7);
  return n.__e = m, u;
}
function rn(e, t, n, i, r) {
  var o, a, s, u, d, _ = n.length, l = _, c = 0;
  for (e.__k = new Array(r), o = 0; o < r; o++) (a = t[o]) != null && typeof a != "boolean" && typeof a != "function" ? (u = o + c, (a = e.__k[o] = typeof a == "string" || typeof a == "number" || typeof a == "bigint" || a.constructor == String ? ue(null, a, null, null, null) : me(a) ? ue(Y, { children: a }, null, null, null) : a.constructor === void 0 && a.__b > 0 ? ue(a.type, a.props, a.key, a.ref ? a.ref : null, a.__v) : a).__ = e, a.__b = e.__b + 1, s = null, (d = a.__i = on(a, n, u, l)) !== -1 && (l--, (s = n[d]) && (s.__u |= 2)), s == null || s.__v === null ? (d == -1 && c--, typeof a.type != "function" && (a.__u |= 4)) : d != u && (d == u - 1 ? c-- : d == u + 1 ? c++ : (d > u ? c-- : c++, a.__u |= 4))) : e.__k[o] = null;
  if (l) for (o = 0; o < _; o++) (s = n[o]) != null && !(2 & s.__u) && (s.__e == i && (i = z(s)), St(s, s));
  return i;
}
function Pt(e, t, n) {
  var i, r;
  if (typeof e.type == "function") {
    for (i = e.__k, r = 0; i && r < i.length; r++) i[r] && (i[r].__ = e, t = Pt(i[r], t, n));
    return t;
  }
  e.__e != t && (t && e.type && !n.contains(t) && (t = z(e)), n.insertBefore(e.__e, t || null), t = e.__e);
  do
    t = t && t.nextSibling;
  while (t != null && t.nodeType == 8);
  return t;
}
function fe(e, t) {
  return t = t || [], e == null || typeof e == "boolean" || (me(e) ? e.some(function(n) {
    fe(n, t);
  }) : t.push(e)), t;
}
function on(e, t, n, i) {
  var r, o, a = e.key, s = e.type, u = t[n];
  if (u === null || u && a == u.key && s === u.type && !(2 & u.__u)) return n;
  if (i > (u != null && !(2 & u.__u) ? 1 : 0)) for (r = n - 1, o = n + 1; r >= 0 || o < t.length; ) {
    if (r >= 0) {
      if ((u = t[r]) && !(2 & u.__u) && a == u.key && s === u.type) return r;
      r--;
    }
    if (o < t.length) {
      if ((u = t[o]) && !(2 & u.__u) && a == u.key && s === u.type) return o;
      o++;
    }
  }
  return -1;
}
function Ue(e, t, n) {
  t[0] == "-" ? e.setProperty(t, n ?? "") : e[t] = n == null ? "" : typeof n != "number" || nn.test(t) ? n : n + "px";
}
function ae(e, t, n, i, r) {
  var o;
  e: if (t == "style") if (typeof n == "string") e.style.cssText = n;
  else {
    if (typeof i == "string" && (e.style.cssText = i = ""), i) for (t in i) n && t in n || Ue(e.style, t, "");
    if (n) for (t in n) i && n[t] === i[t] || Ue(e.style, t, n[t]);
  }
  else if (t[0] == "o" && t[1] == "n") o = t != (t = t.replace(mt, "$1")), t = t.toLowerCase() in e || t == "onFocusOut" || t == "onFocusIn" ? t.toLowerCase().slice(2) : t.slice(2), e.l || (e.l = {}), e.l[t + o] = n, n ? i ? n.u = i.u : (n.u = We, e.addEventListener(t, o ? Ye : we, o)) : e.removeEventListener(t, o ? Ye : we, o);
  else {
    if (r == "http://www.w3.org/2000/svg") t = t.replace(/xlink(H|:h)/, "h").replace(/sName$/, "s");
    else if (t != "width" && t != "height" && t != "href" && t != "list" && t != "form" && t != "tabIndex" && t != "download" && t != "rowSpan" && t != "colSpan" && t != "role" && t != "popover" && t in e) try {
      e[t] = n ?? "";
      break e;
    } catch {
    }
    typeof n == "function" || (n == null || n === !1 && t[4] != "-" ? e.removeAttribute(t) : e.setAttribute(t, t == "popover" && n == 1 ? "" : n));
  }
}
function He(e) {
  return function(t) {
    if (this.l) {
      var n = this.l[t.type + e];
      if (t.t == null) t.t = We++;
      else if (t.t < n.u) return;
      return n(h.event ? h.event(t) : t);
    }
  };
}
function Fe(e, t, n, i, r, o, a, s, u, d) {
  var _, l, c, f, M, m, S, D, y, Z, W, re, q, Ve, oe, ye, Me, E = t.type;
  if (t.constructor !== void 0) return null;
  128 & n.__u && (u = !!(32 & n.__u), o = [s = t.__e = n.__e]), (_ = h.__b) && _(t);
  e: if (typeof E == "function") try {
    if (D = t.props, y = "prototype" in E && E.prototype.render, Z = (_ = E.contextType) && i[_.__c], W = _ ? Z ? Z.props.value : _.__ : i, n.__c ? S = (l = t.__c = n.__c).__ = l.__E : (y ? t.__c = l = new E(D, W) : (t.__c = l = new A(D, W), l.constructor = E, l.render = sn), Z && Z.sub(l), l.props = D, l.state || (l.state = {}), l.context = W, l.__n = i, c = l.__d = !0, l.__h = [], l._sb = []), y && l.__s == null && (l.__s = l.state), y && E.getDerivedStateFromProps != null && (l.__s == l.state && (l.__s = $({}, l.__s)), $(l.__s, E.getDerivedStateFromProps(D, l.__s))), f = l.props, M = l.state, l.__v = t, c) y && E.getDerivedStateFromProps == null && l.componentWillMount != null && l.componentWillMount(), y && l.componentDidMount != null && l.__h.push(l.componentDidMount);
    else {
      if (y && E.getDerivedStateFromProps == null && D !== f && l.componentWillReceiveProps != null && l.componentWillReceiveProps(D, W), !l.__e && (l.shouldComponentUpdate != null && l.shouldComponentUpdate(D, l.__s, W) === !1 || t.__v == n.__v)) {
        for (t.__v != n.__v && (l.props = D, l.state = l.__s, l.__d = !1), t.__e = n.__e, t.__k = n.__k, t.__k.some(function(X) {
          X && (X.__ = t);
        }), re = 0; re < l._sb.length; re++) l.__h.push(l._sb[re]);
        l._sb = [], l.__h.length && a.push(l);
        break e;
      }
      l.componentWillUpdate != null && l.componentWillUpdate(D, l.__s, W), y && l.componentDidUpdate != null && l.__h.push(function() {
        l.componentDidUpdate(f, M, m);
      });
    }
    if (l.context = W, l.props = D, l.__P = e, l.__e = !1, q = h.__r, Ve = 0, y) {
      for (l.state = l.__s, l.__d = !1, q && q(t), _ = l.render(l.props, l.state, l.context), oe = 0; oe < l._sb.length; oe++) l.__h.push(l._sb[oe]);
      l._sb = [];
    } else do
      l.__d = !1, q && q(t), _ = l.render(l.props, l.state, l.context), l.state = l.__s;
    while (l.__d && ++Ve < 25);
    l.state = l.__s, l.getChildContext != null && (i = $($({}, i), l.getChildContext())), y && !c && l.getSnapshotBeforeUpdate != null && (m = l.getSnapshotBeforeUpdate(f, M)), s = Mt(e, me(ye = _ != null && _.type === Y && _.key == null ? _.props.children : _) ? ye : [ye], t, n, i, r, o, a, s, u, d), l.base = t.__e, t.__u &= -161, l.__h.length && a.push(l), S && (l.__E = l.__ = null);
  } catch (X) {
    if (t.__v = null, u || o != null) if (X.then) {
      for (t.__u |= u ? 160 : 128; s && s.nodeType == 8 && s.nextSibling; ) s = s.nextSibling;
      o[o.indexOf(s)] = null, t.__e = s;
    } else for (Me = o.length; Me--; ) $e(o[Me]);
    else t.__e = n.__e, t.__k = n.__k;
    h.__e(X, t, n);
  }
  else o == null && t.__v == n.__v ? (t.__k = n.__k, t.__e = n.__e) : s = t.__e = an(n.__e, t, n, i, r, o, a, u, d);
  return (_ = h.diffed) && _(t), 128 & t.__u ? void 0 : s;
}
function gt(e, t, n) {
  for (var i = 0; i < n.length; i++) Re(n[i], n[++i], n[++i]);
  h.__c && h.__c(t, e), e.some(function(r) {
    try {
      e = r.__h, r.__h = [], e.some(function(o) {
        o.call(r);
      });
    } catch (o) {
      h.__e(o, r.__v);
    }
  });
}
function an(e, t, n, i, r, o, a, s, u) {
  var d, _, l, c, f, M, m, S = n.props, D = t.props, y = t.type;
  if (y == "svg" ? r = "http://www.w3.org/2000/svg" : y == "math" ? r = "http://www.w3.org/1998/Math/MathML" : r || (r = "http://www.w3.org/1999/xhtml"), o != null) {
    for (d = 0; d < o.length; d++) if ((f = o[d]) && "setAttribute" in f == !!y && (y ? f.localName == y : f.nodeType == 3)) {
      e = f, o[d] = null;
      break;
    }
  }
  if (e == null) {
    if (y == null) return document.createTextNode(D);
    e = document.createElementNS(r, y, D.is && D), s && (h.__m && h.__m(t, o), s = !1), o = null;
  }
  if (y === null) S === D || s && e.data === D || (e.data = D);
  else {
    if (o = o && ve.call(e.childNodes), S = n.props || te, !s && o != null) for (S = {}, d = 0; d < e.attributes.length; d++) S[(f = e.attributes[d]).name] = f.value;
    for (d in S) if (f = S[d], d != "children") {
      if (d == "dangerouslySetInnerHTML") l = f;
      else if (!(d in D)) {
        if (d == "value" && "defaultValue" in D || d == "checked" && "defaultChecked" in D) continue;
        ae(e, d, null, f, r);
      }
    }
    for (d in D) f = D[d], d == "children" ? c = f : d == "dangerouslySetInnerHTML" ? _ = f : d == "value" ? M = f : d == "checked" ? m = f : s && typeof f != "function" || S[d] === f || ae(e, d, f, S[d], r);
    if (_) s || l && (_.__html === l.__html || _.__html === e.innerHTML) || (e.innerHTML = _.__html), t.__k = [];
    else if (l && (e.innerHTML = ""), Mt(e, me(c) ? c : [c], t, n, i, y == "foreignObject" ? "http://www.w3.org/1999/xhtml" : r, o, a, o ? o[0] : n.__k && z(n, 0), s, u), o != null) for (d = o.length; d--; ) $e(o[d]);
    s || (d = "value", y == "progress" && M == null ? e.removeAttribute("value") : M !== void 0 && (M !== e[d] || y == "progress" && !M || y == "option" && M !== S[d]) && ae(e, d, M, S[d], r), d = "checked", m !== void 0 && m !== e[d] && ae(e, d, m, S[d], r));
  }
  return e;
}
function Re(e, t, n) {
  try {
    if (typeof e == "function") {
      var i = typeof e.__u == "function";
      i && e.__u(), i && t == null || (e.__u = e(t));
    } else e.current = t;
  } catch (r) {
    h.__e(r, n);
  }
}
function St(e, t, n) {
  var i, r;
  if (h.unmount && h.unmount(e), (i = e.ref) && (i.current && i.current !== e.__e || Re(i, null, t)), (i = e.__c) != null) {
    if (i.componentWillUnmount) try {
      i.componentWillUnmount();
    } catch (o) {
      h.__e(o, t);
    }
    i.base = i.__P = null;
  }
  if (i = e.__k) for (r = 0; r < i.length; r++) i[r] && St(i[r], t, n || typeof e.type != "function");
  n || $e(e.__e), e.__c = e.__ = e.__e = void 0;
}
function sn(e, t, n) {
  return this.constructor(e, n);
}
function Ce(e, t, n) {
  var i, r, o, a;
  t == document && (t = document.documentElement), h.__ && h.__(e, t), r = (i = !1) ? null : t.__k, o = [], a = [], Fe(t, e = t.__k = H(Y, null, [e]), r || te, te, t.namespaceURI, r ? null : t.firstChild ? ve.call(t.childNodes) : null, o, r ? r.__e : t.firstChild, i, a), gt(o, e, a);
}
function ln(e, t) {
  var n = { __c: t = "__cC" + kt++, __: e, Consumer: function(i, r) {
    return i.children(r);
  }, Provider: function(i) {
    var r, o;
    return this.getChildContext || (r = /* @__PURE__ */ new Set(), (o = {})[t] = this, this.getChildContext = function() {
      return o;
    }, this.componentWillUnmount = function() {
      r = null;
    }, this.shouldComponentUpdate = function(a) {
      this.props.value !== a.value && r.forEach(function(s) {
        s.__e = !0, Ne(s);
      });
    }, this.sub = function(a) {
      r.add(a);
      var s = a.componentWillUnmount;
      a.componentWillUnmount = function() {
        r && r.delete(a), s && s.call(a);
      };
    }), i.children;
  } };
  return n.Provider.__ = n.Consumer.contextType = n;
}
ve = Dt.slice, h = { __e: function(e, t, n, i) {
  for (var r, o, a; t = t.__; ) if ((r = t.__c) && !r.__) try {
    if ((o = r.constructor) && o.getDerivedStateFromError != null && (r.setState(o.getDerivedStateFromError(e)), a = r.__d), r.componentDidCatch != null && (r.componentDidCatch(e, i || {}), a = r.__d), a) return r.__E = r;
  } catch (s) {
    e = s;
  }
  throw e;
} }, ht = 0, pt = function(e) {
  return e != null && e.constructor == null;
}, A.prototype.setState = function(e, t) {
  var n;
  n = this.__s != null && this.__s !== this.state ? this.__s : this.__s = $({}, this.state), typeof e == "function" && (e = e($({}, n), this.props)), e && $(n, e), e != null && this.__v && (t && this._sb.push(t), Ne(this));
}, A.prototype.forceUpdate = function(e) {
  this.__v && (this.__e = !0, e && this.__h.push(e), Ne(this));
}, A.prototype.render = Y, V = [], vt = typeof Promise == "function" ? Promise.prototype.then.bind(Promise.resolve()) : setTimeout, be = function(e, t) {
  return e.__v.__b - t.__v.__b;
}, _e.__r = 0, mt = /(PointerCapture)$|Capture$/i, We = 0, we = He(!1), Ye = He(!0), kt = 0;
var un = 0;
function p(e, t, n, i, r, o) {
  t || (t = {});
  var a, s, u = t;
  if ("ref" in u) for (s in u = {}, t) s == "ref" ? a = t[s] : u[s] = t[s];
  var d = { type: e, props: u, key: n, ref: a, __k: null, __: null, __b: 0, __e: null, __c: null, constructor: void 0, __v: --un, __i: -1, __u: 0, __source: r, __self: o };
  if (typeof e == "function" && (a = e.defaultProps)) for (s in a) u[s] === void 0 && (u[s] = a[s]);
  return h.vnode && h.vnode(d), d;
}
var B, P, Pe, ze, Te = 0, bt = [], g = h, Be = g.__b, Ge = g.__r, Je = g.diffed, Ze = g.__c, qe = g.unmount, Xe = g.__;
function ke(e, t) {
  g.__h && g.__h(P, e, Te || t), Te = 0;
  var n = P.__H || (P.__H = { __: [], __h: [] });
  return e >= n.__.length && n.__.push({}), n.__[e];
}
function R(e) {
  return Te = 1, dn(Yt, e);
}
function dn(e, t, n) {
  var i = ke(B++, 2);
  if (i.t = e, !i.__c && (i.__ = [n ? n(t) : Yt(void 0, t), function(s) {
    var u = i.__N ? i.__N[0] : i.__[0], d = i.t(u, s);
    u !== d && (i.__N = [d, i.__[1]], i.__c.setState({}));
  }], i.__c = P, !P.u)) {
    var r = function(s, u, d) {
      if (!i.__c.__H) return !0;
      var _ = i.__c.__H.__.filter(function(c) {
        return !!c.__c;
      });
      if (_.every(function(c) {
        return !c.__N;
      })) return !o || o.call(this, s, u, d);
      var l = i.__c.props !== s;
      return _.forEach(function(c) {
        if (c.__N) {
          var f = c.__[0];
          c.__ = c.__N, c.__N = void 0, f !== c.__[0] && (l = !0);
        }
      }), o && o.call(this, s, u, d) || l;
    };
    P.u = !0;
    var o = P.shouldComponentUpdate, a = P.componentWillUpdate;
    P.componentWillUpdate = function(s, u, d) {
      if (this.__e) {
        var _ = o;
        o = void 0, r(s, u, d), o = _;
      }
      a && a.call(this, s, u, d);
    }, P.shouldComponentUpdate = r;
  }
  return i.__N || i.__;
}
function L(e, t) {
  var n = ke(B++, 3);
  !g.__s && wt(n.__H, t) && (n.__ = e, n.i = t, P.__H.__h.push(n));
}
function je(e, t) {
  var n = ke(B++, 7);
  return wt(n.__H, t) && (n.__ = e(), n.__H = t, n.__h = e), n.__;
}
function I(e) {
  var t = P.context[e.__c], n = ke(B++, 9);
  return n.c = e, t ? (n.__ == null && (n.__ = !0, t.sub(P)), t.props.value) : e.__;
}
function cn() {
  for (var e; e = bt.shift(); ) if (e.__P && e.__H) try {
    e.__H.__h.forEach(de), e.__H.__h.forEach(xe), e.__H.__h = [];
  } catch (t) {
    e.__H.__h = [], g.__e(t, e.__v);
  }
}
g.__b = function(e) {
  P = null, Be && Be(e);
}, g.__ = function(e, t) {
  e && t.__k && t.__k.__m && (e.__m = t.__k.__m), Xe && Xe(e, t);
}, g.__r = function(e) {
  Ge && Ge(e), B = 0;
  var t = (P = e.__c).__H;
  t && (Pe === P ? (t.__h = [], P.__h = [], t.__.forEach(function(n) {
    n.__N && (n.__ = n.__N), n.i = n.__N = void 0;
  })) : (t.__h.forEach(de), t.__h.forEach(xe), t.__h = [], B = 0)), Pe = P;
}, g.diffed = function(e) {
  Je && Je(e);
  var t = e.__c;
  t && t.__H && (t.__H.__h.length && (bt.push(t) !== 1 && ze === g.requestAnimationFrame || ((ze = g.requestAnimationFrame) || _n)(cn)), t.__H.__.forEach(function(n) {
    n.i && (n.__H = n.i), n.i = void 0;
  })), Pe = P = null;
}, g.__c = function(e, t) {
  t.some(function(n) {
    try {
      n.__h.forEach(de), n.__h = n.__h.filter(function(i) {
        return !i.__ || xe(i);
      });
    } catch (i) {
      t.some(function(r) {
        r.__h && (r.__h = []);
      }), t = [], g.__e(i, n.__v);
    }
  }), Ze && Ze(e, t);
}, g.unmount = function(e) {
  qe && qe(e);
  var t, n = e.__c;
  n && n.__H && (n.__H.__.forEach(function(i) {
    try {
      de(i);
    } catch (r) {
      t = r;
    }
  }), n.__H = void 0, t && g.__e(t, n.__v));
};
var Qe = typeof requestAnimationFrame == "function";
function _n(e) {
  var t, n = function() {
    clearTimeout(i), Qe && cancelAnimationFrame(t), setTimeout(e);
  }, i = setTimeout(n, 100);
  Qe && (t = requestAnimationFrame(n));
}
function de(e) {
  var t = P, n = e.__c;
  typeof n == "function" && (e.__c = void 0, n()), P = t;
}
function xe(e) {
  var t = P;
  e.__c = e.__(), P = t;
}
function wt(e, t) {
  return !e || e.length !== t.length || t.some(function(n, i) {
    return n !== e[i];
  });
}
function Yt(e, t) {
  return typeof t == "function" ? t(e) : t;
}
function fn(e, t) {
  for (var n in t) e[n] = t[n];
  return e;
}
function et(e, t) {
  for (var n in e) if (n !== "__source" && !(n in t)) return !0;
  for (var i in t) if (i !== "__source" && e[i] !== t[i]) return !0;
  return !1;
}
function tt(e, t) {
  this.props = e, this.context = t;
}
(tt.prototype = new A()).isPureReactComponent = !0, tt.prototype.shouldComponentUpdate = function(e, t) {
  return et(this.props, e) || et(this.state, t);
};
var nt = h.__b;
h.__b = function(e) {
  e.type && e.type.__f && e.ref && (e.props.ref = e.ref, e.ref = null), nt && nt(e);
};
var hn = h.__e;
h.__e = function(e, t, n, i) {
  if (e.then) {
    for (var r, o = t; o = o.__; ) if ((r = o.__c) && r.__c) return t.__e == null && (t.__e = n.__e, t.__k = n.__k), r.__c(e, t);
  }
  hn(e, t, n, i);
};
var it = h.unmount;
function Nt(e, t, n) {
  return e && (e.__c && e.__c.__H && (e.__c.__H.__.forEach(function(i) {
    typeof i.__c == "function" && i.__c();
  }), e.__c.__H = null), (e = fn({}, e)).__c != null && (e.__c.__P === n && (e.__c.__P = t), e.__c = null), e.__k = e.__k && e.__k.map(function(i) {
    return Nt(i, t, n);
  })), e;
}
function Ct(e, t, n) {
  return e && n && (e.__v = null, e.__k = e.__k && e.__k.map(function(i) {
    return Ct(i, t, n);
  }), e.__c && e.__c.__P === t && (e.__e && n.appendChild(e.__e), e.__c.__e = !0, e.__c.__P = n)), e;
}
function ge() {
  this.__u = 0, this.o = null, this.__b = null;
}
function Tt(e) {
  var t = e.__.__c;
  return t && t.__a && t.__a(e);
}
function se() {
  this.i = null, this.l = null;
}
h.unmount = function(e) {
  var t = e.__c;
  t && t.__R && t.__R(), t && 32 & e.__u && (e.type = null), it && it(e);
}, (ge.prototype = new A()).__c = function(e, t) {
  var n = t.__c, i = this;
  i.o == null && (i.o = []), i.o.push(n);
  var r = Tt(i.__v), o = !1, a = function() {
    o || (o = !0, n.__R = null, r ? r(s) : s());
  };
  n.__R = a;
  var s = function() {
    if (!--i.__u) {
      if (i.state.__a) {
        var u = i.state.__a;
        i.__v.__k[0] = Ct(u, u.__c.__P, u.__c.__O);
      }
      var d;
      for (i.setState({ __a: i.__b = null }); d = i.o.pop(); ) d.forceUpdate();
    }
  };
  i.__u++ || 32 & t.__u || i.setState({ __a: i.__b = i.__v.__k[0] }), e.then(a, a);
}, ge.prototype.componentWillUnmount = function() {
  this.o = [];
}, ge.prototype.render = function(e, t) {
  if (this.__b) {
    if (this.__v.__k) {
      var n = document.createElement("div"), i = this.__v.__k[0].__c;
      this.__v.__k[0] = Nt(this.__b, n, i.__O = i.__P);
    }
    this.__b = null;
  }
  var r = t.__a && H(Y, null, e.fallback);
  return r && (r.__u &= -33), [H(Y, null, t.__a ? null : e.children), r];
};
var rt = function(e, t, n) {
  if (++n[1] === n[0] && e.l.delete(t), e.props.revealOrder && (e.props.revealOrder[0] !== "t" || !e.l.size)) for (n = e.i; n; ) {
    for (; n.length > 3; ) n.pop()();
    if (n[1] < n[0]) break;
    e.i = n = n[2];
  }
};
function pn(e) {
  return this.getChildContext = function() {
    return e.context;
  }, e.children;
}
function vn(e) {
  var t = this, n = e.h;
  t.componentWillUnmount = function() {
    Ce(null, t.v), t.v = null, t.h = null;
  }, t.h && t.h !== n && t.componentWillUnmount(), t.v || (t.h = n, t.v = { nodeType: 1, parentNode: n, childNodes: [], contains: function() {
    return !0;
  }, appendChild: function(i) {
    this.childNodes.push(i), t.h.appendChild(i);
  }, insertBefore: function(i, r) {
    this.childNodes.push(i), t.h.insertBefore(i, r);
  }, removeChild: function(i) {
    this.childNodes.splice(this.childNodes.indexOf(i) >>> 1, 1), t.h.removeChild(i);
  } }), Ce(H(pn, { context: t.context }, e.__v), t.v);
}
function mn(e, t) {
  var n = H(vn, { __v: e, h: t });
  return n.containerInfo = t, n;
}
(se.prototype = new A()).__a = function(e) {
  var t = this, n = Tt(t.__v), i = t.l.get(e);
  return i[0]++, function(r) {
    var o = function() {
      t.props.revealOrder ? (i.push(r), rt(t, e, i)) : r();
    };
    n ? n(o) : o();
  };
}, se.prototype.render = function(e) {
  this.i = null, this.l = /* @__PURE__ */ new Map();
  var t = fe(e.children);
  e.revealOrder && e.revealOrder[0] === "b" && t.reverse();
  for (var n = t.length; n--; ) this.l.set(t[n], this.i = [1, 0, this.i]);
  return e.children;
}, se.prototype.componentDidUpdate = se.prototype.componentDidMount = function() {
  var e = this;
  this.l.forEach(function(t, n) {
    rt(e, n, t);
  });
};
var kn = typeof Symbol < "u" && Symbol.for && Symbol.for("react.element") || 60103, Dn = /^(?:accent|alignment|arabic|baseline|cap|clip(?!PathU)|color|dominant|fill|flood|font|glyph(?!R)|horiz|image(!S)|letter|lighting|marker(?!H|W|U)|overline|paint|pointer|shape|stop|strikethrough|stroke|text(?!L)|transform|underline|unicode|units|v|vector|vert|word|writing|x(?!C))[A-Z]/, yn = /^on(Ani|Tra|Tou|BeforeInp|Compo)/, Mn = /[A-Z0-9]/g, Pn = typeof document < "u", gn = function(e) {
  return (typeof Symbol < "u" && typeof Symbol() == "symbol" ? /fil|che|rad/ : /fil|che|ra/).test(e);
};
A.prototype.isReactComponent = {}, ["componentWillMount", "componentWillReceiveProps", "componentWillUpdate"].forEach(function(e) {
  Object.defineProperty(A.prototype, e, { configurable: !0, get: function() {
    return this["UNSAFE_" + e];
  }, set: function(t) {
    Object.defineProperty(this, e, { configurable: !0, writable: !0, value: t });
  } });
});
var ot = h.event;
function Sn() {
}
function bn() {
  return this.cancelBubble;
}
function wn() {
  return this.defaultPrevented;
}
h.event = function(e) {
  return ot && (e = ot(e)), e.persist = Sn, e.isPropagationStopped = bn, e.isDefaultPrevented = wn, e.nativeEvent = e;
};
var Yn = { enumerable: !1, configurable: !0, get: function() {
  return this.class;
} }, at = h.vnode;
h.vnode = function(e) {
  typeof e.type == "string" && function(t) {
    var n = t.props, i = t.type, r = {}, o = i.indexOf("-") === -1;
    for (var a in n) {
      var s = n[a];
      if (!(a === "value" && "defaultValue" in n && s == null || Pn && a === "children" && i === "noscript" || a === "class" || a === "className")) {
        var u = a.toLowerCase();
        a === "defaultValue" && "value" in n && n.value == null ? a = "value" : a === "download" && s === !0 ? s = "" : u === "translate" && s === "no" ? s = !1 : u[0] === "o" && u[1] === "n" ? u === "ondoubleclick" ? a = "ondblclick" : u !== "onchange" || i !== "input" && i !== "textarea" || gn(n.type) ? u === "onfocus" ? a = "onfocusin" : u === "onblur" ? a = "onfocusout" : yn.test(a) && (a = u) : u = a = "oninput" : o && Dn.test(a) ? a = a.replace(Mn, "-$&").toLowerCase() : s === null && (s = void 0), u === "oninput" && r[a = u] && (a = "oninputCapture"), r[a] = s;
      }
    }
    i == "select" && r.multiple && Array.isArray(r.value) && (r.value = fe(n.children).forEach(function(d) {
      d.props.selected = r.value.indexOf(d.props.value) != -1;
    })), i == "select" && r.defaultValue != null && (r.value = fe(n.children).forEach(function(d) {
      d.props.selected = r.multiple ? r.defaultValue.indexOf(d.props.value) != -1 : r.defaultValue == d.props.value;
    })), n.class && !n.className ? (r.class = n.class, Object.defineProperty(r, "className", Yn)) : (n.className && !n.class || n.class && n.className) && (r.class = r.className = n.className), t.props = r;
  }(e), e.$$typeof = kn, at && at(e);
};
var st = h.__r;
h.__r = function(e) {
  st && st(e), e.__c;
};
var lt = h.diffed;
h.diffed = function(e) {
  lt && lt(e);
  var t = e.props, n = e.__e;
  n != null && e.type === "textarea" && "value" in t && t.value !== n.value && (n.value = t.value == null ? "" : t.value);
};
var Nn = Symbol.for("preact-signals");
function De() {
  if (F > 1)
    F--;
  else {
    for (var e, t = !1; ee !== void 0; ) {
      var n = ee;
      for (ee = void 0, Ee++; n !== void 0; ) {
        var i = n.o;
        if (n.o = void 0, n.f &= -3, !(8 & n.f) && Et(n)) try {
          n.c();
        } catch (r) {
          t || (e = r, t = !0);
        }
        n = i;
      }
    }
    if (Ee = 0, F--, t) throw e;
  }
}
function Cn(e) {
  if (F > 0) return e();
  F++;
  try {
    return e();
  } finally {
    De();
  }
}
var k = void 0, ee = void 0, F = 0, Ee = 0, he = 0;
function xt(e) {
  if (k !== void 0) {
    var t = e.n;
    if (t === void 0 || t.t !== k)
      return t = { i: 0, S: e, p: k.s, n: void 0, t: k, e: void 0, x: void 0, r: t }, k.s !== void 0 && (k.s.n = t), k.s = t, e.n = t, 32 & k.f && e.S(t), t;
    if (t.i === -1)
      return t.i = 0, t.n !== void 0 && (t.n.p = t.p, t.p !== void 0 && (t.p.n = t.n), t.p = k.s, t.n = void 0, k.s.n = t, k.s = t), t;
  }
}
function b(e) {
  this.v = e, this.i = 0, this.n = void 0, this.t = void 0;
}
b.prototype.brand = Nn;
b.prototype.h = function() {
  return !0;
};
b.prototype.S = function(e) {
  this.t !== e && e.e === void 0 && (e.x = this.t, this.t !== void 0 && (this.t.e = e), this.t = e);
};
b.prototype.U = function(e) {
  if (this.t !== void 0) {
    var t = e.e, n = e.x;
    t !== void 0 && (t.x = n, e.e = void 0), n !== void 0 && (n.e = t, e.x = void 0), e === this.t && (this.t = n);
  }
};
b.prototype.subscribe = function(e) {
  var t = this;
  return U(function() {
    var n = t.value, i = k;
    k = void 0;
    try {
      e(n);
    } finally {
      k = i;
    }
  });
};
b.prototype.valueOf = function() {
  return this.value;
};
b.prototype.toString = function() {
  return this.value + "";
};
b.prototype.toJSON = function() {
  return this.value;
};
b.prototype.peek = function() {
  var e = k;
  k = void 0;
  try {
    return this.value;
  } finally {
    k = e;
  }
};
Object.defineProperty(b.prototype, "value", { get: function() {
  var e = xt(this);
  return e !== void 0 && (e.i = this.i), this.v;
}, set: function(e) {
  if (e !== this.v) {
    if (Ee > 100) throw new Error("Cycle detected");
    this.v = e, this.i++, he++, F++;
    try {
      for (var t = this.t; t !== void 0; t = t.x) t.t.N();
    } finally {
      De();
    }
  }
} });
function w(e) {
  return new b(e);
}
function Et(e) {
  for (var t = e.s; t !== void 0; t = t.n) if (t.S.i !== t.i || !t.S.h() || t.S.i !== t.i) return !0;
  return !1;
}
function At(e) {
  for (var t = e.s; t !== void 0; t = t.n) {
    var n = t.S.n;
    if (n !== void 0 && (t.r = n), t.S.n = t, t.i = -1, t.n === void 0) {
      e.s = t;
      break;
    }
  }
}
function Lt(e) {
  for (var t = e.s, n = void 0; t !== void 0; ) {
    var i = t.p;
    t.i === -1 ? (t.S.U(t), i !== void 0 && (i.n = t.n), t.n !== void 0 && (t.n.p = i)) : n = t, t.S.n = t.r, t.r !== void 0 && (t.r = void 0), t = i;
  }
  e.s = n;
}
function G(e) {
  b.call(this, void 0), this.x = e, this.s = void 0, this.g = he - 1, this.f = 4;
}
(G.prototype = new b()).h = function() {
  if (this.f &= -3, 1 & this.f) return !1;
  if ((36 & this.f) == 32 || (this.f &= -5, this.g === he)) return !0;
  if (this.g = he, this.f |= 1, this.i > 0 && !Et(this))
    return this.f &= -2, !0;
  var e = k;
  try {
    At(this), k = this;
    var t = this.x();
    (16 & this.f || this.v !== t || this.i === 0) && (this.v = t, this.f &= -17, this.i++);
  } catch (n) {
    this.v = n, this.f |= 16, this.i++;
  }
  return k = e, Lt(this), this.f &= -2, !0;
};
G.prototype.S = function(e) {
  if (this.t === void 0) {
    this.f |= 36;
    for (var t = this.s; t !== void 0; t = t.n) t.S.S(t);
  }
  b.prototype.S.call(this, e);
};
G.prototype.U = function(e) {
  if (this.t !== void 0 && (b.prototype.U.call(this, e), this.t === void 0)) {
    this.f &= -33;
    for (var t = this.s; t !== void 0; t = t.n) t.S.U(t);
  }
};
G.prototype.N = function() {
  if (!(2 & this.f)) {
    this.f |= 6;
    for (var e = this.t; e !== void 0; e = e.x) e.t.N();
  }
};
Object.defineProperty(G.prototype, "value", { get: function() {
  if (1 & this.f) throw new Error("Cycle detected");
  var e = xt(this);
  if (this.h(), e !== void 0 && (e.i = this.i), 16 & this.f) throw this.v;
  return this.v;
} });
function ut(e) {
  return new G(e);
}
function Ot(e) {
  var t = e.u;
  if (e.u = void 0, typeof t == "function") {
    F++;
    var n = k;
    k = void 0;
    try {
      t();
    } catch (i) {
      throw e.f &= -2, e.f |= 8, Ie(e), i;
    } finally {
      k = n, De();
    }
  }
}
function Ie(e) {
  for (var t = e.s; t !== void 0; t = t.n) t.S.U(t);
  e.x = void 0, e.s = void 0, Ot(e);
}
function Tn(e) {
  if (k !== this) throw new Error("Out-of-order effect");
  Lt(this), k = e, this.f &= -2, 8 & this.f && Ie(this), De();
}
function ne(e) {
  this.x = e, this.u = void 0, this.s = void 0, this.o = void 0, this.f = 32;
}
ne.prototype.c = function() {
  var e = this.S();
  try {
    if (8 & this.f || this.x === void 0) return;
    var t = this.x();
    typeof t == "function" && (this.u = t);
  } finally {
    e();
  }
};
ne.prototype.S = function() {
  if (1 & this.f) throw new Error("Cycle detected");
  this.f |= 1, this.f &= -9, Ot(this), At(this), F++;
  var e = k;
  return k = this, Tn.bind(this, e);
};
ne.prototype.N = function() {
  2 & this.f || (this.f |= 2, this.o = ee, ee = this);
};
ne.prototype.d = function() {
  this.f |= 8, 1 & this.f || Ie(this);
};
function U(e) {
  var t = new ne(e);
  try {
    t.c();
  } catch (n) {
    throw t.d(), n;
  }
  return t.d.bind(t);
}
var Wt, Se, $t = [];
U(function() {
  Wt = this.N;
})();
function J(e, t) {
  h[e] = t.bind(null, h[e] || function() {
  });
}
function pe(e) {
  Se && Se(), Se = e && e.S();
}
function Ft(e) {
  var t = this, n = e.data, i = En(n);
  i.value = n;
  var r = je(function() {
    for (var s = t, u = t.__v; u = u.__; ) if (u.__c) {
      u.__c.__$f |= 4;
      break;
    }
    var d = ut(function() {
      var f = i.value.value;
      return f === 0 ? 0 : f === !0 ? "" : f || "";
    }), _ = ut(function() {
      return !pt(d.value);
    }), l = U(function() {
      if (this.N = Rt, _.value) {
        var f = d.value;
        s.__v && s.__v.__e && s.__v.__e.nodeType === 3 && (s.__v.__e.data = f);
      }
    }), c = t.__$u.d;
    return t.__$u.d = function() {
      l(), c.call(this);
    }, [_, d];
  }, []), o = r[0], a = r[1];
  return o.value ? a.peek() : a.value;
}
Ft.displayName = "_st";
Object.defineProperties(b.prototype, { constructor: { configurable: !0, value: void 0 }, type: { configurable: !0, value: Ft }, props: { configurable: !0, get: function() {
  return { data: this };
} }, __b: { configurable: !0, value: 1 } });
J("__b", function(e, t) {
  if (typeof t.type == "string") {
    var n, i = t.props;
    for (var r in i) if (r !== "children") {
      var o = i[r];
      o instanceof b && (n || (t.__np = n = {}), n[r] = o, i[r] = o.peek());
    }
  }
  e(t);
});
J("__r", function(e, t) {
  pe();
  var n, i = t.__c;
  i && (i.__$f &= -2, (n = i.__$u) === void 0 && (i.__$u = n = function(r) {
    var o;
    return U(function() {
      o = this;
    }), o.c = function() {
      i.__$f |= 1, i.setState({});
    }, o;
  }())), pe(n), e(t);
});
J("__e", function(e, t, n, i) {
  pe(), e(t, n, i);
});
J("diffed", function(e, t) {
  pe();
  var n;
  if (typeof t.type == "string" && (n = t.__e)) {
    var i = t.__np, r = t.props;
    if (i) {
      var o = n.U;
      if (o) for (var a in o) {
        var s = o[a];
        s !== void 0 && !(a in i) && (s.d(), o[a] = void 0);
      }
      else
        o = {}, n.U = o;
      for (var u in i) {
        var d = o[u], _ = i[u];
        d === void 0 ? (d = xn(n, u, _, r), o[u] = d) : d.o(_, r);
      }
    }
  }
  e(t);
});
function xn(e, t, n, i) {
  var r = t in e && e.ownerSVGElement === void 0, o = w(n);
  return { o: function(a, s) {
    o.value = a, i = s;
  }, d: U(function() {
    this.N = Rt;
    var a = o.value.value;
    i[t] !== a && (i[t] = a, r ? e[t] = a : a ? e.setAttribute(t, a) : e.removeAttribute(t));
  }) };
}
J("unmount", function(e, t) {
  if (typeof t.type == "string") {
    var n = t.__e;
    if (n) {
      var i = n.U;
      if (i) {
        n.U = void 0;
        for (var r in i) {
          var o = i[r];
          o && o.d();
        }
      }
    }
  } else {
    var a = t.__c;
    if (a) {
      var s = a.__$u;
      s && (a.__$u = void 0, s.d());
    }
  }
  e(t);
});
J("__h", function(e, t, n, i) {
  (i < 3 || i === 9) && (t.__$f |= 2), e(t, n, i);
});
A.prototype.shouldComponentUpdate = function(e, t) {
  var n = this.__$u, i = n && n.s !== void 0;
  for (var r in t) return !0;
  if (this.__f || typeof this.u == "boolean" && this.u === !0) {
    var o = 2 & this.__$f;
    if (!(i || o || 4 & this.__$f) || 1 & this.__$f) return !0;
  } else if (!(i || 4 & this.__$f) || 3 & this.__$f) return !0;
  for (var a in e) if (a !== "__source" && e[a] !== this.props[a]) return !0;
  for (var s in this.props) if (!(s in e)) return !0;
  return !1;
};
function En(e) {
  return je(function() {
    return w(e);
  }, []);
}
var An = function(e) {
  queueMicrotask(function() {
    queueMicrotask(e);
  });
};
function Ln() {
  Cn(function() {
    for (var e; e = $t.shift(); ) Wt.call(e);
  });
}
function Rt() {
  $t.push(this) === 1 && (h.requestAnimationFrame || An)(Ln);
}
const O = ln({}), dt = {
  DATE_STRING: /^\d{4}-\d{2}-\d{2}$/,
  DATE_TIME_STRING: /^\d{4}-\d{2}-\d{2} \d{2}:\d{2}$/
};
class On extends Error {
  constructor(t) {
    super(`Invalid date time specification: ${t}`);
  }
}
const j = (e) => {
  if (!dt.DATE_TIME_STRING.test(e) && !dt.DATE_STRING.test(e))
    throw new On(e);
  return new Date(
    Number(e.slice(0, 4)),
    Number(e.slice(5, 7)) - 1,
    Number(e.slice(8, 10)),
    Number(e.slice(11, 13)),
    // for date strings this will be 0
    Number(e.slice(14, 16))
    // for date strings this will be 0
  );
}, ie = (e) => {
  const t = e.slice(11, 13), n = e.slice(14, 16);
  return {
    year: Number(e.slice(0, 4)),
    month: Number(e.slice(5, 7)) - 1,
    date: Number(e.slice(8, 10)),
    hours: t !== "" ? Number(t) : void 0,
    minutes: n !== "" ? Number(n) : void 0
  };
}, jt = (e, t) => e.toLocaleString(t, { month: "long" }), It = (e, t) => e.toLocaleString(t, {
  month: "numeric",
  day: "numeric",
  year: "numeric"
}), Wn = (e, t) => e.map((n) => n.toLocaleString(t, { weekday: "short" }).charAt(0)), $n = (e, t) => t === "he-IL" ? e.toLocaleString(t, { weekday: "narrow" }) : e.toLocaleString(t, { weekday: "short" }), Fn = (e, t) => e.map((n) => $n(n, t)), Rn = (e, t) => ["zh-cn", "zh-tw", "ca-es", "he-il"].includes(t.toLowerCase()) ? Fn(e, t) : Wn(e, t);
var jn = "data:image/svg+xml,%3c%3fxml version='1.0' encoding='utf-8'%3f%3e%3c!-- Uploaded to: SVG Repo%2c www.svgrepo.com%2c Generator: SVG Repo Mixer Tools --%3e%3csvg width='800px' height='800px' viewBox='0 0 24 24' fill='none' xmlns='http://www.w3.org/2000/svg'%3e%3cpath d='M6 9L12 15L18 9' stroke='%23DED8E1' stroke-width='4' stroke-linecap='round' stroke-linejoin='round'/%3e%3c/svg%3e";
const ce = () => "s" + Math.random().toString(36).substring(2, 11), Vt = (e) => e.key === "Enter" || e.key === " ";
function In() {
  const e = ce(), t = ce(), n = ce(), i = I(O), r = (c) => c === "" ? i.translate("MM/DD/YYYY") : It(j(c), i.config.locale.value);
  L(() => {
    i.datePickerState.inputDisplayedValue.value = r(i.datePickerState.selectedDate.value);
  }, [i.datePickerState.selectedDate.value, i.config.locale.value]);
  const [o, a] = R([]), s = () => {
    const c = document.getElementById(n);
    i.datePickerState.inputWrapperElement.value = c instanceof HTMLDivElement ? c : void 0;
  };
  L(() => {
    i.config.teleportTo && s();
    const c = ["sx__date-input-wrapper"];
    i.datePickerState.isOpen.value && c.push("sx__date-input--active"), a(c);
  }, [i.datePickerState.isOpen.value]);
  const u = (c) => {
    c.key === "Enter" && d(c);
  }, d = (c) => {
    c.stopPropagation();
    try {
      i.datePickerState.inputDisplayedValue.value = c.target.value, i.datePickerState.close();
    } catch (f) {
      console.log("Error setting input value:" + f);
    }
  };
  L(() => {
    const c = document.getElementById(e);
    if (c !== null)
      return c.addEventListener("change", d), () => c.removeEventListener("change", d);
  });
  const _ = (c) => {
    d(c), i.datePickerState.open();
  }, l = (c) => {
    Vt(c) && (c.preventDefault(), i.datePickerState.open(), setTimeout(() => {
      const f = document.querySelector('[data-focus="true"]');
      f instanceof HTMLElement && f.focus();
    }, 50));
  };
  return p(Y, { children: p("div", { className: o.join(" "), id: n, children: [p("label", { for: e, id: t, className: "sx__date-input-label", children: i.config.label || i.translate("Date") }), p("input", { id: e, tabIndex: i.datePickerState.isDisabled.value ? -1 : 0, name: i.config.name || "date", "aria-describedby": t, value: i.datePickerState.inputDisplayedValue.value, "data-testid": "date-picker-input", className: "sx__date-input", onClick: _, onKeyUp: u, type: "text" }), p("button", { type: "button", tabIndex: i.datePickerState.isDisabled.value ? -1 : 0, "aria-label": i.translate("Choose Date"), onKeyDown: l, onClick: () => i.datePickerState.open(), className: "sx__date-input-chevron-wrapper", children: p("img", { className: "sx__date-input-chevron", src: jn, alt: "" }) })] }) });
}
var K;
(function(e) {
  e.MONTH_DAYS = "month-days", e.YEARS = "years";
})(K || (K = {}));
const Vn = "years-view", Kn = "months-view", Un = "date-picker-week";
class Hn extends Error {
  constructor(t, n) {
    super(`Number must be between ${t} and ${n}.`), Object.defineProperty(this, "min", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: t
    }), Object.defineProperty(this, "max", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: n
    });
  }
}
const N = (e) => {
  if (e < 0 || e > 99)
    throw new Hn(0, 99);
  return String(e).padStart(2, "0");
}, x = (e) => `${e.getFullYear()}-${N(e.getMonth() + 1)}-${N(e.getDate())}`, zn = (e) => `${N(e.getHours())}:${N(e.getMinutes())}`, Kt = (e) => `${x(e)} ${zn(e)}`, Ut = (e, t) => {
  const { year: n, month: i, date: r, hours: o, minutes: a } = ie(e), s = o !== void 0 && a !== void 0, u = new Date(n, i, r, o ?? 0, a ?? 0);
  let d = (u.getMonth() + t) % 12;
  return d < 0 && (d += 12), u.setMonth(u.getMonth() + t), u.getMonth() > d ? u.setDate(0) : u.getMonth() < d && (u.setMonth(u.getMonth() + 1), u.setDate(0)), s ? Kt(u) : x(u);
}, Bn = (e, t) => {
  const { year: n, month: i, date: r, hours: o, minutes: a } = ie(e), s = o !== void 0 && a !== void 0, u = new Date(n, i, r, o ?? 0, a ?? 0);
  return u.setDate(u.getDate() + t), s ? Kt(u) : x(u);
}, Ht = (e, t) => (e = e.slice(0, 8) + N(t) + e.slice(10), e), Gn = (e) => (e = Ut(e, -1), Ht(e, 1)), Jn = (e) => (e = Ut(e, 1), Ht(e, 1));
function ct({ direction: e, onClick: t, buttonText: n, disabled: i = !1 }) {
  return p("button", { type: "button", disabled: i, className: "sx__chevron-wrapper sx__ripple", onMouseUp: t, onKeyDown: (o) => {
    Vt(o) && t();
  }, tabIndex: 0, children: p("i", { className: `sx__chevron sx__chevron--${e}`, children: n }) });
}
function Zn({ setYearsView: e }) {
  const t = I(O), n = (l) => {
    const c = j(l);
    return jt(c, t.config.locale.value);
  }, i = (l) => ie(l).year, [r, o] = R(n(t.datePickerState.datePickerDate.value)), [a, s] = R(i(t.datePickerState.datePickerDate.value)), u = () => {
    t.datePickerState.datePickerDate.value = Gn(t.datePickerState.datePickerDate.value);
  }, d = () => {
    t.datePickerState.datePickerDate.value = Jn(t.datePickerState.datePickerDate.value);
  };
  L(() => {
    o(n(t.datePickerState.datePickerDate.value)), s(i(t.datePickerState.datePickerDate.value));
  }, [t.datePickerState.datePickerDate.value]);
  const _ = (l) => {
    l.stopPropagation(), e();
  };
  return p(Y, { children: p("header", { className: "sx__date-picker__month-view-header", children: [p(ct, { direction: "previous", onClick: () => u(), buttonText: t.translate("Previous month") }), p("button", { type: "button", className: "sx__date-picker__month-view-header__month-year", onClick: (l) => _(l), children: r + " " + a }), p(ct, { direction: "next", onClick: () => d(), buttonText: t.translate("Next month") })] }) });
}
function qn() {
  const e = I(O), t = e.timeUnitsImpl.getWeekFor(j(e.datePickerState.datePickerDate.value)), n = Rn(t, e.config.locale.value);
  return p("div", { className: "sx__date-picker__day-names", children: n.map((i) => p("span", { "data-testid": "day-name", className: "sx__date-picker__day-name", children: i })) });
}
const Xn = (e) => {
  const t = /* @__PURE__ */ new Date();
  return e.getDate() === t.getDate() && e.getMonth() === t.getMonth() && e.getFullYear() === t.getFullYear();
}, Qn = (e, t) => e.getMonth() === t.getMonth() && e.getFullYear() === t.getFullYear();
var Ae;
(function(e) {
  e[e.SUNDAY = 0] = "SUNDAY", e[e.MONDAY = 1] = "MONDAY", e[e.TUESDAY = 2] = "TUESDAY", e[e.WEDNESDAY = 3] = "WEDNESDAY", e[e.THURSDAY = 4] = "THURSDAY", e[e.FRIDAY = 5] = "FRIDAY", e[e.SATURDAY = 6] = "SATURDAY";
})(Ae || (Ae = {}));
const ei = "en-US", ti = Ae.MONDAY, ni = (e, t) => {
  const { year: n, month: i, date: r } = ie(e);
  return new Date(n, i, r).toLocaleDateString(t, {
    day: "numeric",
    month: "long",
    year: "numeric"
  });
}, ii = ni;
function ri({ week: e }) {
  const t = I(O), n = e.map((s) => {
    const u = ["sx__date-picker__day"];
    return Xn(s) && u.push("sx__date-picker__day--today"), x(s) === t.datePickerState.selectedDate.value && u.push("sx__date-picker__day--selected"), Qn(s, j(t.datePickerState.datePickerDate.value)) || u.push("is-leading-or-trailing"), {
      day: s,
      classes: u
    };
  }), i = (s) => {
    const u = x(s);
    return u >= t.config.min && u <= t.config.max;
  }, r = (s) => {
    t.datePickerState.selectedDate.value = x(s), t.datePickerState.close();
  }, o = (s) => x(s.day) === t.datePickerState.datePickerDate.value, a = (s) => {
    if (s.key === "Enter") {
      t.datePickerState.selectedDate.value = t.datePickerState.datePickerDate.value, t.datePickerState.close();
      return;
    }
    const u = /* @__PURE__ */ new Map([
      ["ArrowDown", 7],
      ["ArrowUp", -7],
      ["ArrowLeft", -1],
      ["ArrowRight", 1]
    ]);
    t.datePickerState.datePickerDate.value = Bn(t.datePickerState.datePickerDate.value, u.get(s.key) || 0);
  };
  return p(Y, { children: p("div", { "data-testid": Un, className: "sx__date-picker__week", children: n.map((s) => p("button", { type: "button", tabIndex: o(s) ? 0 : -1, disabled: !i(s.day), "aria-label": ii(t.datePickerState.datePickerDate.value, t.config.locale.value), className: s.classes.join(" "), "data-focus": o(s) ? "true" : void 0, onClick: () => r(s.day), onKeyDown: a, children: s.day.getDate() })) }) });
}
function oi({ seatYearsView: e }) {
  const t = ce(), n = I(O), [i, r] = R([]), o = () => {
    const a = j(n.datePickerState.datePickerDate.value);
    r(n.timeUnitsImpl.getMonthWithTrailingAndLeadingDays(a.getFullYear(), a.getMonth()));
  };
  return L(() => {
    o();
  }, [n.datePickerState.datePickerDate.value]), L(() => {
    const a = new MutationObserver((u) => {
      u.forEach((d) => {
        const _ = d.target;
        _.dataset.focus === "true" && _.focus();
      });
    }), s = document.getElementById(t);
    return a.observe(s, {
      childList: !0,
      subtree: !0,
      attributes: !0
    }), () => a.disconnect();
  }, []), p(Y, { children: p("div", { id: t, "data-testid": Kn, className: "sx__date-picker__month-view", children: [p(Zn, { setYearsView: e }), p(qn, {}), i.map((a) => p(ri, { week: a }))] }) });
}
function ai({ year: e, setYearAndMonth: t, isExpanded: n, expand: i }) {
  const r = I(O), o = r.timeUnitsImpl.getMonthsFor(e), a = (s, u) => {
    s.stopPropagation(), t(e, u.getMonth());
  };
  return p(Y, { children: p("li", { className: n ? "sx__is-expanded" : "", children: [p("button", { type: "button", className: "sx__date-picker__years-accordion__expand-button sx__ripple--wide", onClick: () => i(e), children: e }), n && p("div", { className: "sx__date-picker__years-view-accordion__panel", children: o.map((s) => p("button", { type: "button", className: "sx__date-picker__years-view-accordion__month", onClick: (u) => a(u, s), children: jt(s, r.config.locale.value) })) })] }) });
}
function si({ setMonthView: e }) {
  const t = I(O), n = j(t.config.min).getFullYear(), i = j(t.config.max).getFullYear(), r = Array.from({ length: i - n + 1 }, (d, _) => n + _), { year: o } = ie(t.datePickerState.selectedDate.value), [a, s] = R(o), u = (d, _) => {
    t.datePickerState.datePickerDate.value = x(new Date(d, _, 1)), e();
  };
  return L(() => {
    var d;
    const _ = (d = document.querySelector(".sx__date-picker__years-view")) === null || d === void 0 ? void 0 : d.querySelector(".sx__is-expanded");
    _ && _.scrollIntoView({
      block: "center"
    });
  }, []), p(Y, { children: p("ul", { className: "sx__date-picker__years-view", "data-testid": Vn, children: r.map((d) => p(ai, { year: d, setYearAndMonth: (_, l) => u(_, l), isExpanded: a === d, expand: (_) => s(_) })) }) });
}
const li = (e) => {
  if (e) {
    const t = e.scrollHeight > e.clientHeight, i = window.getComputedStyle(e).overflowY.indexOf("hidden") !== -1;
    return t && !i;
  }
  return !0;
}, zt = (e, t = []) => !e || e === document.body || e.nodeType === Node.DOCUMENT_FRAGMENT_NODE ? (t.push(window), t) : (li(e) && t.push(e), zt(e.assignedSlot ? e.assignedSlot.parentNode : e.parentNode, t)), _t = "sx__date-picker-popup";
function ui() {
  const e = I(O), [t, n] = R(K.MONTH_DAYS), i = je(() => {
    const c = [
      _t,
      e.datePickerState.isDark.value ? "is-dark" : "",
      e.config.teleportTo ? "is-teleported" : ""
    ];
    return e.config.placement && !e.config.teleportTo && c.push(e.config.placement), c;
  }, [
    e.datePickerState.isDark.value,
    e.config.placement,
    e.config.teleportTo
  ]), r = (c) => {
    c.target.closest(`.${_t}`) || e.datePickerState.close();
  }, o = (c) => {
    c.key === "Escape" && (e.config.listeners.onEscapeKeyDown ? e.config.listeners.onEscapeKeyDown(e) : e.datePickerState.close());
  };
  L(() => (document.addEventListener("click", r), document.addEventListener("keydown", o), () => {
    document.removeEventListener("click", r), document.removeEventListener("keydown", o);
  }), []);
  const a = Number(getComputedStyle(document.documentElement).fontSize.split("px")[0]), s = 362, u = 332, d = () => {
    const c = e.datePickerState.inputWrapperElement.value, f = c == null ? void 0 : c.getBoundingClientRect();
    if (!(c === void 0 || !(f instanceof DOMRect)))
      return {
        top: e.config.placement.includes("bottom") ? f.height + f.y + 1 : f.y - a - s,
        // subtract remsize to leave room for label text
        left: e.config.placement.includes("start") ? f.x : f.x + f.width - u,
        width: u,
        position: "fixed"
      };
  }, [_, l] = R(d());
  return L(() => {
    const c = e.datePickerState.inputWrapperElement.value;
    if (c === void 0)
      return;
    const f = zt(c), M = () => l(d());
    return f.forEach((m) => m.addEventListener("scroll", M)), () => f.forEach((m) => m.removeEventListener("scroll", M));
  }, []), p(Y, { children: p("div", { style: e.config.teleportTo ? _ : void 0, "data-testid": "date-picker-popup", className: i.join(" "), children: t === K.MONTH_DAYS ? p(oi, { seatYearsView: () => n(K.YEARS) }) : p(si, { setMonthView: () => n(K.MONTH_DAYS) }) }) });
}
function di({ $app: e }) {
  const t = ["sx__date-picker-wrapper"], [n, i] = R(t);
  L(() => {
    var o;
    const a = [...t];
    e.datePickerState.isDark.value && a.push("is-dark"), !((o = e.config.style) === null || o === void 0) && o.fullWidth && a.push("has-full-width"), e.datePickerState.isDisabled.value && a.push("is-disabled"), i(a);
  }, [e.datePickerState.isDark.value, e.datePickerState.isDisabled.value]);
  let r = p(ui, {});
  return e.config.teleportTo && (r = mn(r, e.config.teleportTo)), p(Y, { children: p("div", { className: n.join(" "), children: p(O.Provider, { value: e, children: [p(In, {}), e.datePickerState.isOpen.value && r] }) }) });
}
class ci {
  constructor(t) {
    Object.defineProperty(this, "$app", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: t
    });
  }
  render(t) {
    Ce(H(di, {
      $app: this.$app
    }), t);
  }
  get value() {
    return this.$app.datePickerState.selectedDate.value;
  }
  set value(t) {
    this.$app.datePickerState.selectedDate.value = t;
  }
  get disabled() {
    return this.$app.datePickerState.isDisabled.value;
  }
  set disabled(t) {
    this.$app.datePickerState.isDisabled.value = t;
  }
  setTheme(t) {
    this.$app.datePickerState.isDark.value = t === "dark";
  }
  getTheme() {
    return this.$app.datePickerState.isDark.value ? "dark" : "light";
  }
}
var Le;
(function(e) {
  e[e.JANUARY = 0] = "JANUARY", e[e.FEBRUARY = 1] = "FEBRUARY", e[e.MARCH = 2] = "MARCH", e[e.APRIL = 3] = "APRIL", e[e.MAY = 4] = "MAY", e[e.JUNE = 5] = "JUNE", e[e.JULY = 6] = "JULY", e[e.AUGUST = 7] = "AUGUST", e[e.SEPTEMBER = 8] = "SEPTEMBER", e[e.OCTOBER = 9] = "OCTOBER", e[e.NOVEMBER = 10] = "NOVEMBER", e[e.DECEMBER = 11] = "DECEMBER";
})(Le || (Le = {}));
class Bt extends Error {
  constructor() {
    super("Year zero does not exist in the Gregorian calendar.");
  }
}
class _i extends Date {
  constructor(t, n, i) {
    if (super(t, n, i), t === 0)
      throw new Bt();
    this.setFullYear(t);
  }
  get year() {
    return this.getFullYear();
  }
  get month() {
    return this.getMonth();
  }
  get date() {
    return this.getDate();
  }
}
class fi {
  constructor(t) {
    Object.defineProperty(this, "config", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: t
    });
  }
  get firstDayOfWeek() {
    return this.config.firstDayOfWeek.value;
  }
  set firstDayOfWeek(t) {
    this.config.firstDayOfWeek.value = t;
  }
  getMonthWithTrailingAndLeadingDays(t, n) {
    if (t === 0)
      throw new Bt();
    const i = new Date(t, n, 1), r = [this.getWeekFor(i)];
    let o = !0, a = r[0][0];
    for (; o; ) {
      const s = new Date(a.getFullYear(), a.getMonth(), a.getDate() + 7);
      s.getMonth() === n ? (r.push(this.getWeekFor(s)), a = s) : o = !1;
    }
    return r;
  }
  getWeekFor(t) {
    const n = [this.getFirstDateOfWeek(t)];
    for (; n.length < 7; ) {
      const i = n[n.length - 1], r = new Date(i);
      r.setDate(i.getDate() + 1), n.push(r);
    }
    return n;
  }
  getMonthsFor(t) {
    return Object.values(Le).filter((n) => !isNaN(Number(n))).map((n) => new _i(t, Number(n), 1));
  }
  getFirstDateOfWeek(t) {
    const n = t.getDay() - this.firstDayOfWeek, i = t;
    return n === 0 || (n > 0 ? i.setDate(t.getDate() - n) : i.setDate(t.getDate() - (7 + n))), i;
  }
}
class hi {
  constructor() {
    Object.defineProperty(this, "config", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    });
  }
  build() {
    return new fi(this.config);
  }
  withConfig(t) {
    return this.config = t, this;
  }
}
var C;
(function(e) {
  e.SLASH = "/", e.DASH = "-", e.PERIOD = ".";
})(C || (C = {}));
var T;
(function(e) {
  e.DMY = "DMY", e.MDY = "MDY", e.YMD = "YMD";
})(T || (T = {}));
const v = {
  slashMDY: {
    delimiter: C.SLASH,
    order: T.MDY
  },
  slashDMY: {
    delimiter: C.SLASH,
    order: T.DMY
  },
  slashYMD: {
    delimiter: C.SLASH,
    order: T.YMD
  },
  periodDMY: {
    delimiter: C.PERIOD,
    order: T.DMY
  },
  dashYMD: {
    delimiter: C.DASH,
    order: T.YMD
  },
  dashDMY: {
    delimiter: C.DASH,
    order: T.DMY
  }
}, pi = /* @__PURE__ */ new Map([
  ["ca-ES", v.slashDMY],
  ["cs-CZ", v.periodDMY],
  ["da-DK", v.periodDMY],
  ["de-DE", v.periodDMY],
  ["en-GB", v.slashDMY],
  ["en-US", v.slashMDY],
  ["es-ES", v.slashDMY],
  ["et-EE", v.periodDMY],
  ["fi-FI", v.periodDMY],
  ["fr-FR", v.slashDMY],
  ["fr-CH", v.periodDMY],
  ["hr-HR", v.periodDMY],
  ["id-ID", v.slashDMY],
  ["it-IT", v.slashDMY],
  ["ja-JP", v.slashYMD],
  ["ko-KR", v.slashYMD],
  ["ky-KG", v.slashDMY],
  ["lt-LT", v.dashYMD],
  ["mk-MK", v.periodDMY],
  ["nl-NL", v.dashDMY],
  ["pl-PL", v.periodDMY],
  ["pt-BR", v.slashDMY],
  ["ro-RO", v.periodDMY],
  ["ru-RU", v.periodDMY],
  ["sk-SK", v.periodDMY],
  ["sl-SI", v.periodDMY],
  ["sr-Latn-RS", v.periodDMY],
  ["sr-RS", v.periodDMY],
  ["sv-SE", v.dashYMD],
  ["tr-TR", v.periodDMY],
  ["uk-UA", v.periodDMY],
  ["zh-CN", v.slashYMD],
  ["zh-TW", v.slashYMD]
]);
class vi extends Error {
  constructor(t) {
    super(`Locale not supported: ${t}`);
  }
}
class Gt extends Error {
  constructor(t, n) {
    super(`Invalid date format: ${t} for locale: ${n}`);
  }
}
const le = (e, t, n) => {
  const i = e.match(t);
  if (!i)
    throw new Gt(e, n);
  return i;
}, mi = (e, t) => {
  if (/^\d{4}-\d{2}-\d{2}$/.test(e))
    return e;
  const i = pi.get(t);
  if (!i)
    throw new vi(t);
  const { order: r, delimiter: o } = i, a = /^(\d{1,2})\/(\d{1,2})\/(\d{4})$/, s = /^(\d{1,2})\.(\d{1,2})\.(\d{4})$/, u = /^(\d{4})\/(\d{1,2})\/(\d{1,2})$/;
  if (r === T.DMY && o === C.SLASH) {
    const d = le(e, a, t), [, _, l, c] = d;
    return `${c}-${N(+l)}-${N(+_)}`;
  }
  if (r === T.MDY && o === C.SLASH) {
    const d = le(e, a, t), [, _, l, c] = d;
    return `${c}-${N(+_)}-${N(+l)}`;
  }
  if (r === T.YMD && o === C.SLASH) {
    const d = le(e, u, t), [, _, l, c] = d;
    return `${_}-${N(+l)}-${N(+c)}`;
  }
  if (r === T.DMY && o === C.PERIOD) {
    const d = le(e, s, t), [, _, l, c] = d;
    return `${c}-${N(+l)}-${N(+_)}`;
  }
  throw new Gt(e, t);
}, ki = (e, t) => {
  var n;
  const i = x(/* @__PURE__ */ new Date()), r = typeof t == "string" ? t : i, o = w(!1), a = w(e.disabled || !1), s = w(K.MONTH_DAYS), u = w(r), d = w(r || i), _ = w(((n = e.style) === null || n === void 0 ? void 0 : n.dark) || !1), l = w(t ? It(j(t), e.locale.value) : ""), c = w(l.value);
  U(() => {
    try {
      const m = mi(l.value, e.locale.value);
      if (m < e.min || m > e.max) {
        l.value = c.value;
        return;
      }
      u.value = m, d.value = m, c.value = l.value;
    } catch {
    }
  });
  let f = !1;
  const M = (m) => {
    if (!f)
      return f = !0;
    e.listeners.onChange(m);
  };
  return U(() => {
    var m;
    !((m = e.listeners) === null || m === void 0) && m.onChange && M(u.value);
  }), {
    inputWrapperElement: w(void 0),
    isOpen: o,
    isDisabled: a,
    datePickerView: s,
    selectedDate: u,
    datePickerDate: d,
    inputDisplayedValue: l,
    isDark: _,
    open: () => o.value = !0,
    close: () => o.value = !1,
    toggle: () => o.value = !o.value,
    setView: (m) => s.value = m
  };
};
class Di {
  constructor(t, n, i, r) {
    Object.defineProperty(this, "datePickerState", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: t
    }), Object.defineProperty(this, "config", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: n
    }), Object.defineProperty(this, "timeUnitsImpl", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: i
    }), Object.defineProperty(this, "translate", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: r
    });
  }
}
class yi {
  constructor() {
    Object.defineProperty(this, "datePickerState", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "config", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "timeUnitsImpl", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "translate", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    });
  }
  build() {
    return new Di(this.datePickerState, this.config, this.timeUnitsImpl, this.translate);
  }
  withDatePickerState(t) {
    return this.datePickerState = t, this;
  }
  withConfig(t) {
    return this.config = t, this;
  }
  withTimeUnitsImpl(t) {
    return this.timeUnitsImpl = t, this;
  }
  withTranslate(t) {
    return this.translate = t, this;
  }
}
var Oe;
(function(e) {
  e.TOP_START = "top-start", e.TOP_END = "top-end", e.BOTTOM_START = "bottom-start", e.BOTTOM_END = "bottom-end";
})(Oe || (Oe = {}));
class Mi {
  constructor(t = ei, n = ti, i = x(new Date(1970, 0, 1)), r = x(new Date((/* @__PURE__ */ new Date()).getFullYear() + 50, 11, 31)), o = Oe.BOTTOM_START, a = {}, s = {}, u, d, _, l) {
    Object.defineProperty(this, "min", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: i
    }), Object.defineProperty(this, "max", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: r
    }), Object.defineProperty(this, "placement", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: o
    }), Object.defineProperty(this, "listeners", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: a
    }), Object.defineProperty(this, "style", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: s
    }), Object.defineProperty(this, "teleportTo", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: u
    }), Object.defineProperty(this, "label", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: d
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: _
    }), Object.defineProperty(this, "disabled", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: l
    }), Object.defineProperty(this, "locale", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "firstDayOfWeek", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), this.locale = w(t), this.firstDayOfWeek = w(n);
  }
}
class Pi {
  constructor() {
    Object.defineProperty(this, "locale", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "firstDayOfWeek", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "min", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "max", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "placement", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "listeners", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "style", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "teleportTo", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "label", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "name", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    }), Object.defineProperty(this, "disabled", {
      enumerable: !0,
      configurable: !0,
      writable: !0,
      value: void 0
    });
  }
  build() {
    return new Mi(this.locale, this.firstDayOfWeek, this.min, this.max, this.placement, this.listeners, this.style, this.teleportTo, this.label, this.name, this.disabled);
  }
  withLocale(t) {
    return this.locale = t, this;
  }
  withFirstDayOfWeek(t) {
    return this.firstDayOfWeek = t, this;
  }
  withMin(t) {
    return this.min = t, this;
  }
  withMax(t) {
    return this.max = t, this;
  }
  withPlacement(t) {
    return this.placement = t, this;
  }
  withListeners(t) {
    return this.listeners = t, this;
  }
  withStyle(t) {
    return this.style = t, this;
  }
  withTeleportTo(t) {
    return this.teleportTo = t, this;
  }
  withLabel(t) {
    return this.label = t, this;
  }
  withName(t) {
    return this.name = t, this;
  }
  withDisabled(t) {
    return this.disabled = t, this;
  }
}
const gi = {
  Date: "",
  "MM/DD/YYYY": "DD/MM/YYYY",
  // Keep format unchanged unless you want to localize it
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, Si = {
  Time: "",
  AM: "",
  PM: "",
  Cancel: "",
  OK: "",
  "Select time": " "
}, bi = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": "  ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "     ",
  "Link to {{n}} more events on {{date}}": "  {{n}}    {{date}}",
  "Link to 1 more event on {{date}}": "     {{date}}",
  CW: " {{week}}"
}, wi = {
  ...bi,
  ...gi,
  ...Si
}, Yi = {
  Date: "Datum",
  "MM/DD/YYYY": "TT.MM.JJJJ",
  "Next month": "Nchster Monat",
  "Previous month": "Vorheriger Monat",
  "Choose Date": "Datum auswhlen"
}, Ni = {
  Today: "Heute",
  Month: "Monat",
  Week: "Woche",
  Day: "Tag",
  "Select View": "Ansicht auswhlen",
  events: "Ereignisse",
  event: "Ereignis",
  "No events": "Keine Ereignisse",
  "Next period": "Nchster Zeitraum",
  "Previous period": "Vorheriger Zeitraum",
  to: "bis",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Ganztgige und mehrtgige Ereignisse",
  "Link to {{n}} more events on {{date}}": "Link zu {{n}} weiteren Ereignissen am {{date}}",
  "Link to 1 more event on {{date}}": "Link zu 1 weiteren Ereignis am {{date}}",
  CW: "KW {{week}}"
}, Ci = {
  Time: "Uhrzeit",
  AM: "AM",
  PM: "PM",
  Cancel: "Abbrechen",
  OK: "OK",
  "Select time": "Uhrzeit auswhlen"
}, Ti = {
  ...Yi,
  ...Ni,
  ...Ci
}, xi = {
  Date: "Date",
  "MM/DD/YYYY": "MM/DD/YYYY",
  "Next month": "Next month",
  "Previous month": "Previous month",
  "Choose Date": "Choose Date"
}, Ei = {
  Today: "Today",
  Month: "Month",
  Week: "Week",
  Day: "Day",
  "Select View": "Select View",
  events: "events",
  event: "event",
  "No events": "No events",
  "Next period": "Next period",
  "Previous period": "Previous period",
  to: "to",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Full day- and multiple day events",
  "Link to {{n}} more events on {{date}}": "Link to {{n}} more events on {{date}}",
  "Link to 1 more event on {{date}}": "Link to 1 more event on {{date}}",
  CW: "Week {{week}}"
}, Ai = {
  Time: "Time",
  AM: "AM",
  PM: "PM",
  Cancel: "Cancel",
  OK: "OK",
  "Select time": "Select time"
}, Li = {
  ...xi,
  ...Ei,
  ...Ai
}, Oi = {
  Date: "Data",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Mese successivo",
  "Previous month": "Mese precedente",
  "Choose Date": "Scegli la data"
}, Wi = {
  Today: "Oggi",
  Month: "Mese",
  Week: "Settimana",
  Day: "Giorno",
  "Select View": "Seleziona la vista",
  events: "eventi",
  event: "evento",
  "No events": "Nessun evento",
  "Next period": "Periodo successivo",
  "Previous period": "Periodo precedente",
  to: "a",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Eventi della giornata e plurigiornalieri",
  "Link to {{n}} more events on {{date}}": "Link a {{n}} eventi in pi il {{date}}",
  "Link to 1 more event on {{date}}": "Link a 1 evento in pi il {{date}}",
  CW: "Settimana {{week}}"
}, $i = {
  Time: "Ora",
  AM: "AM",
  PM: "PM",
  Cancel: "Annulla",
  OK: "OK",
  "Select time": "Seleziona ora"
}, Fi = {
  ...Oi,
  ...Wi,
  ...$i
}, Ri = {
  Date: "Date",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Next month",
  "Previous month": "Previous month",
  "Choose Date": "Choose Date"
}, ji = {
  Today: "Today",
  Month: "Month",
  Week: "Week",
  Day: "Day",
  "Select View": "Select View",
  events: "events",
  event: "event",
  "No events": "No events",
  "Next period": "Next period",
  "Previous period": "Previous period",
  to: "to",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Full day- and multiple day events",
  "Link to {{n}} more events on {{date}}": "Link to {{n}} more events on {{date}}",
  "Link to 1 more event on {{date}}": "Link to 1 more event on {{date}}",
  CW: "Week {{week}}"
}, Ii = {
  Time: "Time",
  AM: "AM",
  PM: "PM",
  Cancel: "Cancel",
  OK: "OK",
  "Select time": "Select time"
}, Vi = {
  ...Ri,
  ...ji,
  ...Ii
}, Ki = {
  Date: "Datum",
  "MM/DD/YYYY": "-MM-DD",
  "Next month": "Nsta mnad",
  "Previous month": "Fregende mnad",
  "Choose Date": "Vlj datum"
}, Ui = {
  Today: "Idag",
  Month: "Mnad",
  Week: "Vecka",
  Day: "Dag",
  "Select View": "Vlj vy",
  events: "hndelser",
  event: "hndelse",
  "No events": "Inga hndelser",
  "Next period": "Nsta period",
  "Previous period": "Fregende period",
  to: "till",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Heldags- och flerdagshndelser",
  "Link to {{n}} more events on {{date}}": "Lnk till {{n}} fler hndelser den {{date}}",
  "Link to 1 more event on {{date}}": "Lnk till 1 hndelse till den {{date}}",
  CW: "Vecka {{week}}"
}, Hi = {
  Time: "Tid",
  AM: "FM",
  PM: "EM",
  Cancel: "Avbryt",
  OK: "OK",
  "Select time": "Vlj tid"
}, zi = {
  ...Ki,
  ...Ui,
  ...Hi
}, Bi = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": "",
  "Previous month": "",
  "Choose Date": ""
}, Gi = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": "",
  events: "",
  event: "",
  "No events": "",
  "Next period": "",
  "Previous period": "",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "",
  "Link to {{n}} more events on {{date}}": "{{date}}{{n}}",
  "Link to 1 more event on {{date}}": "{{date}}1",
  CW: "{{week}}"
}, Ji = {
  Time: "",
  AM: "",
  PM: "",
  Cancel: "",
  OK: "",
  "Select time": ""
}, Zi = {
  ...Bi,
  ...Gi,
  ...Ji
}, qi = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": "",
  "Previous month": "",
  "Choose Date": ""
}, Xi = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": "",
  events: "",
  event: "",
  "No events": "",
  "Next period": "",
  "Previous period": "",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "",
  "Link to {{n}} more events on {{date}}": "{{date}}{{n}}",
  "Link to 1 more event on {{date}}": "{{date}}1",
  CW: "{{week}}"
}, Qi = {
  Time: "",
  AM: "",
  PM: "",
  Cancel: "",
  OK: "",
  "Select time": ""
}, er = {
  ...qi,
  ...Xi,
  ...Qi
}, tr = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": "",
  "Previous month": "",
  "Choose Date": ""
}, nr = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": "",
  events: "",
  event: "",
  "No events": "",
  "Next period": "",
  "Previous period": "",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "",
  "Link to {{n}} more events on {{date}}": "{{date}} {{n}}",
  "Link to 1 more event on {{date}}": "{{date}} 1",
  CW: " {{week}}"
}, ir = {
  Time: "",
  AM: "",
  PM: "",
  Cancel: "",
  OK: "OK",
  "Select time": ""
}, rr = {
  ...tr,
  ...nr,
  ...ir
}, or = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, ar = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "       ",
  "Link to {{n}} more events on {{date}}": "  {{n}}    {{date}}",
  "Link to 1 more event on {{date}}": "  1    {{date}}",
  CW: " {{week}}"
}, sr = {
  Time: "",
  AM: "AM",
  PM: "PM",
  Cancel: "",
  OK: "",
  "Select time": " "
}, lr = {
  ...or,
  ...ar,
  ...sr
}, ur = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, dr = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": "",
  "Previous period": "",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "   ",
  "Link to {{n}} more events on {{date}}": "{{date}} {{n}}   ",
  "Link to 1 more event on {{date}}": "{{date}} 1   ",
  CW: "{{week}}"
}, cr = {
  Time: "",
  AM: "",
  PM: "",
  Cancel: "",
  OK: "",
  "Select time": " "
}, _r = {
  ...ur,
  ...dr,
  ...cr
}, fr = {
  Date: "Date",
  "MM/DD/YYYY": "JJ/MM/AAAA",
  "Next month": "Mois suivant",
  "Previous month": "Mois prcdent",
  "Choose Date": "Choisir une date"
}, hr = {
  Today: "Aujourd'hui",
  Month: "Mois",
  Week: "Semaine",
  Day: "Jour",
  "Select View": "Choisir la vue",
  events: "vnements",
  event: "vnement",
  "No events": "Aucun vnement",
  "Next period": "Priode suivante",
  "Previous period": "Priode prcdente",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "vnements d'une ou plusieurs journes",
  "Link to {{n}} more events on {{date}}": "Lien vers {{n}} autres vnements le {{date}}",
  "Link to 1 more event on {{date}}": "Lien vers 1 autre vnement le {{date}}",
  CW: "Semaine {{week}}"
}, pr = {
  Time: "Heure",
  AM: "AM",
  PM: "PM",
  Cancel: "Annuler",
  OK: "OK",
  "Select time": "Slectionner l'heure"
}, vr = {
  ...fr,
  ...hr,
  ...pr
}, mr = {
  Date: "Dato",
  "MM/DD/YYYY": "-MM-DD",
  "Next month": "Nste mned",
  "Previous month": "Foregende mned",
  "Choose Date": "Vlg dato"
}, kr = {
  Today: "I dag",
  Month: "Mned",
  Week: "Uge",
  Day: "Dag",
  "Select View": "Vlg visning",
  events: "begivenheder",
  event: "begivenhed",
  "No events": "Ingen begivenheder",
  "Next period": "Nste periode",
  "Previous period": "Forgende periode",
  to: "til",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Heldagsbegivenheder og flerdagsbegivenheder",
  "Link to {{n}} more events on {{date}}": "Link til {{n}} flere begivenheder den {{date}}",
  "Link to 1 more event on {{date}}": "Link til 1 mere begivenhed den {{date}}",
  CW: "Uge {{week}}"
}, Dr = {
  Time: "Tid",
  AM: "AM",
  PM: "PM",
  Cancel: "Annuller",
  OK: "OK",
  "Select time": "Vlg tid"
}, yr = {
  ...mr,
  ...kr,
  ...Dr
}, Mr = {
  Date: "Data",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Nastpny miesic",
  "Previous month": "Poprzedni miesic",
  "Choose Date": "Wybiewrz dat"
}, Pr = {
  Today: "Dzisiaj",
  Month: "Miesic",
  Week: "Tydzie",
  Day: "Dzie",
  "Select View": "Wybierz widok",
  events: "wydarzenia",
  event: "wydarzenie",
  "No events": "Brak wydarze",
  "Next period": "Nastpny okres",
  "Previous period": "Poprzedni okres",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Wydarzenia caodniowe i wielodniowe",
  "Link to {{n}} more events on {{date}}": "Link do {{n}} kolejnych wydarze w dniu {{date}}",
  "Link to 1 more event on {{date}}": "Link do 1 kolejnego wydarzenia w dniu {{date}}",
  CW: "Tydzie {{week}}"
}, gr = {
  Time: "Godzina",
  AM: "AM",
  PM: "PM",
  Cancel: "Anuluj",
  OK: "OK",
  "Select time": "Wybierz godzin"
}, Sr = {
  ...Mr,
  ...Pr,
  ...gr
}, br = {
  Date: "Fecha",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Siguiente mes",
  "Previous month": "Mes anterior",
  "Choose Date": "Seleccione una fecha"
}, wr = {
  Today: "Hoy",
  Month: "Mes",
  Week: "Semana",
  Day: "Da",
  "Select View": "Seleccione una vista",
  events: "eventos",
  event: "evento",
  "No events": "Sin eventos",
  "Next period": "Siguiente perodo",
  "Previous period": "Perodo anterior",
  to: "a",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Da completo y eventos de mltiples das",
  "Link to {{n}} more events on {{date}}": "Enlace a {{n}} eventos ms el {{date}}",
  "Link to 1 more event on {{date}}": "Enlace a 1 evento ms el {{date}}",
  CW: "Semana {{week}}"
}, Yr = {
  Time: "Hora",
  AM: "AM",
  PM: "PM",
  Cancel: "Cancelar",
  OK: "Aceptar",
  "Select time": "Seleccionar hora"
}, Nr = {
  ...br,
  ...wr,
  ...Yr
}, Cr = {
  Today: "Vandaag",
  Month: "Maand",
  Week: "Week",
  Day: "Dag",
  "Select View": "Kies weergave",
  events: "gebeurtenissen",
  event: "gebeurtenis",
  "No events": "Geen gebeurtenissen",
  "Next period": "Volgende periode",
  "Previous period": "Vorige periode",
  to: "tot",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Evenementen van een hele dag en meerdere dagen",
  "Link to {{n}} more events on {{date}}": "Link naar {{n}} meer evenementen op {{date}}",
  "Link to 1 more event on {{date}}": "Link naar 1 meer evenement op {{date}}",
  CW: "Week {{week}}"
}, Tr = {
  Date: "Datum",
  "MM/DD/YYYY": "DD-MM-JJJJ",
  "Next month": "Volgende maand",
  "Previous month": "Vorige maand",
  "Choose Date": "Kies datum"
}, xr = {
  Time: "Tijd",
  AM: "AM",
  PM: "PM",
  Cancel: "Annuleren",
  OK: "OK",
  "Select time": "Selecteer tijd"
}, Er = {
  ...Tr,
  ...Cr,
  ...xr
}, Ar = {
  Date: "Data",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Ms seguinte",
  "Previous month": "Ms anterior",
  "Choose Date": "Escolha uma data"
}, Lr = {
  Today: "Hoje",
  Month: "Ms",
  Week: "Semana",
  Day: "Dia",
  "Select View": "Selecione uma visualizao",
  events: "eventos",
  event: "evento",
  "No events": "Sem eventos",
  "Next period": "Perodo seguinte",
  "Previous period": "Perodo anterior",
  to: "a",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Dia inteiro e eventos de vrios dias",
  "Link to {{n}} more events on {{date}}": "Link para mais {{n}} eventos em {{date}}",
  "Link to 1 more event on {{date}}": "Link para mais 1 evento em {{date}}",
  CW: "Semana {{week}}"
}, Or = {
  Time: "Hora",
  AM: "AM",
  PM: "PM",
  Cancel: "Cancelar",
  OK: "OK",
  "Select time": "Selecionar hora"
}, Wr = {
  ...Ar,
  ...Lr,
  ...Or
}, $r = {
  Date: "Dtum",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "al mesiac",
  "Previous month": "Predchdzajci mesiac",
  "Choose Date": "Vyberte dtum"
}, Fr = {
  Today: "Dnes",
  Month: "Mesiac",
  Week: "Tde",
  Day: "De",
  "Select View": "Vyberte zobrazenie",
  events: "udalosti",
  event: "udalos",
  "No events": "iadne udalosti",
  "Next period": "alie obdobie",
  "Previous period": "Predchdzajce obdobie",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Celodenn a viacdov udalosti",
  "Link to {{n}} more events on {{date}}": "Odkaz na {{n}} alch udalost da {{date}}",
  "Link to 1 more event on {{date}}": "Odkaz na 1 aliu udalos da {{date}}",
  CW: "{{week}}. tde"
}, Rr = {
  Time: "as",
  AM: "AM",
  PM: "PM",
  Cancel: "Zrui",
  OK: "OK",
  "Select time": "Vybra as"
}, jr = {
  ...$r,
  ...Fr,
  ...Rr
}, Ir = {
  Date: "",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, Vr = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "   ",
  "Link to {{n}} more events on {{date}}": "  {{n}}    {{date}}",
  "Link to 1 more event on {{date}}": "  1    {{date}}",
  CW: " {{week}}"
}, Kr = {
  Time: "",
  AM: "AM",
  PM: "PM",
  Cancel: "",
  OK: " ",
  "Select time": " "
}, Ur = {
  ...Ir,
  ...Vr,
  ...Kr
}, Hr = {
  Date: "Tarih",
  "MM/DD/YYYY": "GG/AA/YYYY",
  "Next month": "Sonraki ay",
  "Previous month": "nceki ay",
  "Choose Date": "Tarih Se"
}, zr = {
  Today: "Bugn",
  Month: "Aylk",
  Week: "Haftalk",
  Day: "Gnlk",
  "Select View": "Grnm Se",
  events: "etkinlikler",
  event: "etkinlik",
  "No events": "Etkinlik yok",
  "Next period": "Sonraki dnem",
  "Previous period": "nceki dnem",
  to: "dan",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Tm gn ve oklu gn etkinlikleri",
  "Link to {{n}} more events on {{date}}": "{{date}} tarihinde {{n}} etkinlie balant",
  "Link to 1 more event on {{date}}": "{{date}} tarihinde 1 etkinlie balant",
  CW: "{{week}}. Hafta"
}, Br = {
  Time: "Zaman",
  AM: "",
  PM: "S",
  Cancel: "ptal",
  OK: "Tamam",
  "Select time": "Zaman se"
}, Gr = {
  ...Hr,
  ...zr,
  ...Br
}, Jr = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, Zr = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "         ",
  "Link to {{n}} more events on {{date}}": "{{date}}  {{n}}  ",
  "Link to 1 more event on {{date}}": "{{date}}  1  ",
  CW: " {{week}}"
}, qr = {
  Time: "",
  AM: "AM",
  PM: "PM",
  Cancel: "",
  OK: "",
  "Select time": " "
}, Xr = {
  ...Jr,
  ...Zr,
  ...qr
}, Qr = {
  Date: "Tanggal",
  "MM/DD/YYYY": "DD.MM.YYYY",
  "Next month": "Bulan depan",
  "Previous month": "Bulan sebelumnya",
  "Choose Date": "Pilih tanggal"
}, eo = {
  Today: "Hari Ini",
  Month: "Bulan",
  Week: "Minggu",
  Day: "Hari",
  "Select View": "Pilih tampilan",
  events: "Acara",
  event: "Acara",
  "No events": "Tidak ada acara",
  "Next period": "Periode selanjutnya",
  "Previous period": "Periode sebelumnya",
  to: "sampai",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Sepanjang hari dan acara beberapa hari ",
  "Link to {{n}} more events on {{date}}": "Tautan ke {{n}} acara lainnya pada {{date}}",
  "Link to 1 more event on {{date}}": "Tautan ke 1 acara lainnya pada {{date}}",
  CW: "Minggu {{week}}"
}, to = {
  Time: "Waktu",
  AM: "AM",
  PM: "PM",
  Cancel: "Batalkan",
  OK: "OK",
  "Select time": "Pilih waktu"
}, no = {
  ...Qr,
  ...eo,
  ...to
}, io = {
  Date: "Datum",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Dal msc",
  "Previous month": "Pedchoz msc",
  "Choose Date": "Vyberte datum"
}, ro = {
  Today: "Dnes",
  Month: "Msc",
  Week: "Tden",
  Day: "Den",
  "Select View": "Vyberte zobrazen",
  events: "udlosti",
  event: "udlost",
  "No events": "dn udlosti",
  "Next period": "Pt obdob",
  "Previous period": "Pedchoz obdob",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Celodenn a vcedenn udlosti",
  "Link to {{n}} more events on {{date}}": "Odkaz na {{n}} dalch udlost dne {{date}}",
  "Link to 1 more event on {{date}}": "Odkaz na 1 dal udlost dne {{date}}",
  CW: "Tden {{week}}"
}, oo = {
  Time: "as",
  AM: "Dopoledne",
  PM: "Odpoledne",
  Cancel: "Zruit",
  OK: "OK",
  "Select time": "Vyberte as"
}, ao = {
  ...io,
  ...ro,
  ...oo
}, so = {
  Date: "Kuupev",
  "MM/DD/YYYY": "PP.KK.AAAA",
  "Next month": "Jrgmine kuu",
  "Previous month": "Eelmine kuu",
  "Choose Date": "Vali kuupev"
}, lo = {
  Today: "Tna",
  Month: "Kuu",
  Week: "Ndal",
  Day: "Pev",
  "Select View": "Vali vaade",
  events: "sndmused",
  event: "sndmus",
  "No events": "Pole sndmusi",
  "Next period": "Jrgmine periood",
  "Previous period": "Eelmine periood",
  to: "kuni",
  "Full day- and multiple day events": "Tispeva- ja mitmepevasndmused",
  "Link to {{n}} more events on {{date}}": "Link {{n}} rohkematele sndmustele kuupeval {{date}}",
  "Link to 1 more event on {{date}}": "Link hele lisasndmusele kuupeval {{date}}",
  CW: "Ndala number {{week}}"
}, uo = {
  Time: "Aeg",
  AM: "AM",
  PM: "PM",
  Cancel: "Loobu",
  OK: "OK",
  "Select time": "Vali aeg"
}, co = {
  ...so,
  ...lo,
  ...uo
}, _o = {
  Date: "",
  "MM/DD/YYYY": "//",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, fo = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "       ",
  "Link to {{n}} more events on {{date}}": "  {{n}}    {{date}}",
  "Link to 1 more event on {{date}}": "  1    {{date}}",
  CW: " {{week}}"
}, ho = {
  Time: "",
  AM: "AM",
  PM: "PM",
  Cancel: "",
  OK: "",
  "Select time": " "
}, po = {
  ..._o,
  ...fo,
  ...ho
}, vo = {
  Date: "Datum",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Sledei mesec",
  "Previous month": "Prethodni mesec",
  "Choose Date": "Izaberite datum"
}, mo = {
  Today: "Danas",
  Month: "Mesec",
  Week: "Nedelja",
  Day: "Dan",
  "Select View": "Odaberite pregled",
  events: "Dogaaji",
  event: "Dogaaj",
  "No events": "Nema dogaaja",
  "Next period": "Naredni period",
  "Previous period": "Prethodni period",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Celodnevni i viednevni dogaaji",
  "Link to {{n}} more events on {{date}}": "Link do jo {{n}} dogaaja na {{date}}",
  "Link to 1 more event on {{date}}": "Link do jednog dogaaja na {{date}}",
  CW: "Nedelja {{week}}"
}, ko = {
  Time: "Vrijeme",
  AM: "AM",
  PM: "PM",
  Cancel: "Otkai",
  OK: "U redu",
  "Select time": "Odaberi vrijeme"
}, Do = {
  ...vo,
  ...mo,
  ...ko
}, yo = {
  Date: "Data",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Segent mes",
  "Previous month": "Mes anterior",
  "Choose Date": "Selecciona una data"
}, Mo = {
  Today: "Avui",
  Month: "Mes",
  Week: "Setmana",
  Day: "Dia",
  "Select View": "Selecciona una vista",
  events: "Esdeveniments",
  event: "Esdeveniment",
  "No events": "Sense esdeveniments",
  "Next period": "Segent perode",
  "Previous period": "Perode anterior",
  to: "a",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Esdeveniments de dia complet i de mltiples dies",
  "Link to {{n}} more events on {{date}}": "Enlla a {{n}} esdeveniments ms el {{date}}",
  "Link to 1 more event on {{date}}": "Enlla a 1 esdeveniment ms el {{date}}",
  CW: "Setmana {{week}}"
}, Po = {
  Time: "Hora",
  AM: "AM",
  PM: "PM",
  Cancel: "Cancellar",
  OK: "Acceptar",
  "Select time": "Selecciona una hora"
}, go = {
  ...yo,
  ...Mo,
  ...Po
}, So = {
  Date: "",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, bo = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "   ",
  "Link to {{n}} more events on {{date}}": "   {{n}}   {{date}}",
  "Link to 1 more event on {{date}}": "   1  {{date}}",
  CW: " {{week}}"
}, wo = {
  Time: "",
  AM: "AM",
  PM: "PM",
  Cancel: "",
  OK: " ",
  "Select time": " "
}, Yo = {
  ...So,
  ...bo,
  ...wo
}, No = {
  Date: "Data",
  "MM/DD/YYYY": "MMMM-MM-DD",
  "Next month": "Kitas mnuo",
  "Previous month": "Ankstesnis mnuo",
  "Choose Date": "Pasirinkite dat"
}, Co = {
  Today: "iandien",
  Month: "Mnuo",
  Week: "Savait",
  Day: "Diena",
  "Select View": "Pasirinkite vaizd",
  events: "vykiai",
  event: "vykis",
  "No events": "vyki nra",
  "Next period": "Kitas laikotarpis",
  "Previous period": "Ankstesnis laikotarpis",
  to: "iki",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Visos dienos ir keli dien vykiai",
  "Link to {{n}} more events on {{date}}": "Nuoroda  dar {{n}} vykius {{date}}",
  "Link to 1 more event on {{date}}": "Nuoroda  dar 1 vien vyk {{date}}",
  CW: "{{week}} savait"
}, To = {
  Time: "Laikas",
  AM: "AM",
  PM: "PM",
  Cancel: "Ataukti",
  OK: "Gerai",
  "Select time": "Pasirinkite laik"
}, xo = {
  ...No,
  ...Co,
  ...To
}, Eo = {
  Date: "Datum",
  "MM/DD/YYYY": "DD/MM/YYYY",
  "Next month": "Sljedei mjesec",
  "Previous month": "Prethodni mjesec",
  "Choose Date": "Izaberite datum"
}, Ao = {
  Today: "Danas",
  Month: "Mjesec",
  Week: "Nedjelja",
  Day: "Dan",
  "Select View": "Odaberite pregled",
  events: "Dogaaji",
  event: "Dogaaj",
  "No events": "Nema dogaaja",
  "Next period": "Sljedei period",
  "Previous period": "Prethodni period",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Cjelodnevni i viednevni dogaaji",
  "Link to {{n}} more events on {{date}}": "Link do jo {{n}} dogaaja na {{date}}",
  "Link to 1 more event on {{date}}": "Link do jo jednog dogaaja na {{date}}",
  CW: "{{week}}. tjedan"
}, Lo = {
  Time: "Vrijeme",
  AM: "AM",
  PM: "PM",
  Cancel: "Otkai",
  OK: "U redu",
  "Select time": "Odaberi vrijeme"
}, Oo = {
  ...Eo,
  ...Ao,
  ...Lo
}, Wo = {
  Date: "Datum",
  "MM/DD/YYYY": "MM.DD.YYYY",
  "Next month": "Naslednji mesec",
  "Previous month": "Prejnji mesec",
  "Choose Date": "Izberi datum"
}, $o = {
  Today: "Danes",
  Month: "Mesec",
  Week: "Teden",
  Day: "Dan",
  "Select View": "Izberi pogled",
  events: "dogodki",
  event: "dogodek",
  "No events": "Ni dogodkov",
  "Next period": "Naslednji dogodek",
  "Previous period": "Prejnji dogodek",
  to: "do",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Celodnevni in vednevni dogodki",
  "Link to {{n}} more events on {{date}}": "Povezava do {{n}} drugih dogodkov dne {{date}}",
  "Link to 1 more event on {{date}}": "Povezava do e enega dogodka dne {{date}}",
  CW: "Teden {{week}}"
}, Fo = {
  Time: "as",
  AM: "AM",
  PM: "PM",
  Cancel: "Preklii",
  OK: "V redu",
  "Select time": "Izberite as"
}, Ro = {
  ...Wo,
  ...$o,
  ...Fo
}, jo = {
  Date: "Pivmr",
  "MM/DD/YYYY": "VVVV-KK-PP",
  "Next month": "Seuraava kuukausi",
  "Previous month": "Edellinen kuukausi",
  "Choose Date": "Valitse pivmr"
}, Io = {
  Today: "Tnn",
  Month: "Kuukausi",
  Week: "Viikko",
  Day: "Piv",
  "Select View": "Valitse nkym",
  events: "tapahtumaa",
  event: "tapahtuma",
  "No events": "Ei tapahtumia",
  "Next period": "Seuraava ajanjakso",
  "Previous period": "Edellinen ajanjakso",
  to: "-",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Koko ja usean pivn tapahtumat",
  "Link to {{n}} more events on {{date}}": "Linkki {{n}} listapahtumaan pivmrll {{date}}",
  "Link to 1 more event on {{date}}": "Linkki 1 listapahtumaan pivmrll {{date}}",
  CW: "Viikko {{week}}"
}, Vo = {
  Time: "Aika",
  AM: "ap.",
  PM: "ip.",
  Cancel: "Peruuta",
  OK: "OK",
  "Select time": "Valitse aika"
}, Ko = {
  ...jo,
  ...Io,
  ...Vo
}, Uo = {
  Date: "Data",
  "MM/DD/YYYY": "LL/ZZ/AAAA",
  "Next month": "Luna urmtoare",
  "Previous month": "Luna anterioar",
  "Choose Date": "Alege data"
}, Ho = {
  Today: "Astzi",
  Month: "Lun",
  Week: "Sptmn",
  Day: "Zi",
  "Select View": "Selecteaz vizualizarea",
  events: "evenimente",
  event: "eveniment",
  "No events": "Fr evenimente",
  "Next period": "Perioada urmtoare",
  "Previous period": "Perioada anterioar",
  to: "pn la",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "Evenimente pe durata ntregii zile i pe durata mai multor zile",
  "Link to {{n}} more events on {{date}}": "Link ctre {{n}} evenimente suplimentare pe {{date}}",
  "Link to 1 more event on {{date}}": "Link ctre 1 eveniment suplimentar pe {{date}}",
  CW: "Sptmna {{week}}"
}, zo = {
  Time: "Timp",
  AM: "AM",
  PM: "PM",
  Cancel: "Anuleaz",
  OK: "OK",
  "Select time": "Selectai ora"
}, Bo = {
  ...Uo,
  ...Ho,
  ...zo
}, Go = {
  Date: "",
  "MM/DD/YYYY": "MM/DD/YYYY",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, Jo = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": "  ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  "Full day- and multiple day events": "     ",
  "Link to {{n}} more events on {{date}}": "  {{n}}     {{date}}",
  "Link to 1 more event on {{date}}": "  1     {{date}}",
  CW: " {{week}}"
}, Zo = {
  Time: "",
  AM: ".",
  PM: ".",
  Cancel: "",
  OK: "",
  "Select time": " "
}, qo = {
  ...Go,
  ...Jo,
  ...Zo
};
class Xo extends Error {
  constructor(t) {
    super(`Invalid locale: ${t}`);
  }
}
const Qo = (e, t) => (n, i) => {
  if (!/^[a-z]{2}-[A-Z]{2}$/.test(e.value) && e.value !== "sr-Latn-RS")
    throw new Xo(e.value);
  const r = e.value.replaceAll("-", ""), o = t.value[r];
  if (!o)
    return n;
  let a = o[n] || n;
  return Object.keys(i || {}).forEach((s) => {
    const u = String(i == null ? void 0 : i[s]);
    u && (a = a.replace(`{{${s}}}`, u));
  }), a;
}, ea = {
  Date: "",
  "MM/DD/YYYY": "MM/DD/YYYY",
  "Next month": " ",
  "Previous month": " ",
  "Choose Date": " "
}, ta = {
  Today: "",
  Month: "",
  Week: "",
  Day: "",
  "Select View": " ",
  events: "",
  event: "",
  "No events": " ",
  "Next period": " ",
  "Previous period": " ",
  to: "",
  // as in 2/1/2020 to 2/2/2020
  "Full day- and multiple day events": "    ",
  "Link to {{n}} more events on {{date}}": "  {{n}}  -{{date}}",
  "Link to 1 more event on {{date}}": "   -{{date}}",
  CW: "{{week}} "
}, na = {
  Time: "",
  AM: '"',
  PM: '"',
  Cancel: "",
  OK: "",
  "Select time": " "
}, ia = {
  ...ea,
  ...ta,
  ...na
}, ra = {
  deDE: Ti,
  enUS: Li,
  itIT: Fi,
  enGB: Vi,
  svSE: zi,
  zhCN: Zi,
  zhTW: er,
  jaJP: rr,
  ruRU: lr,
  koKR: _r,
  frFR: vr,
  daDK: yr,
  mkMK: Ur,
  plPL: Sr,
  heIL: ia,
  esES: Nr,
  nlNL: Er,
  ptBR: Wr,
  skSK: jr,
  trTR: Gr,
  kyKG: Xr,
  idID: no,
  csCZ: ao,
  etEE: co,
  ukUA: po,
  caES: go,
  srLatnRS: Do,
  srRS: Yo,
  ltLT: xo,
  hrHR: Oo,
  slSI: Ro,
  fiFI: Ko,
  roRO: Bo,
  faIR: qo,
  arEG: wi
}, oa = (e = {}) => {
  const t = new Pi().withFirstDayOfWeek(e.firstDayOfWeek).withLocale(e.locale).withMin(e.min).withMax(e.max).withPlacement(e.placement).withListeners(e.listeners).withStyle(e.style).withTeleportTo(e.teleportTo).withLabel(e.label).withName(e.name).withDisabled(e.disabled).build(), n = new hi().withConfig(t).build();
  return new yi().withConfig(t).withDatePickerState(ki(t, e.selectedDate)).withTimeUnitsImpl(n).withTranslate(Qo(t.locale, w(ra))).build();
}, aa = (e) => {
  const t = oa(e);
  return new ci(t);
}, ua = ft({
  name: "SxDatePicker",
  props: {
    modelValue: {
      type: String,
      required: !0
    },
    config: {
      type: Object,
      default: () => ({
        listeners: {}
      })
    }
  },
  emits: ["update:modelValue"],
  setup(e, { emit: t, expose: n }) {
    const i = "sx" + Math.random().toString(36).substr(2, 9);
    let r;
    return n({
      setTheme: (a) => {
        r == null || r.setTheme(a);
      }
    }), Xt(() => e.modelValue, (a) => {
      (r == null ? void 0 : r.value) !== a && r && (r.value = a);
    }), Qt(() => {
      r = aa({
        ...e.config,
        listeners: {
          ...e.config.listeners,
          onChange: (a) => {
            t("update:modelValue", a);
          }
        },
        selectedDate: e.modelValue
      }), r.render(document.getElementById(i));
    }), () => Q("div", {
      id: i,
      class: "sx-vue-date-picker-wrapper"
    });
  }
});
export {
  la as ScheduleXCalendar,
  ua as SxDatePicker
};
